---
title: "RSS-NET analysis of simulated GWAS summary statistics and B cell regulatory network"
author: "Xiang Zhu"
date: "2019-08-18"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

[Zhu et al (2020)]: https://doi.org/10.1101/2020.03.13.990010
[`wtccc_bcell/`]: https://github.com/SUwonglab/rss-net/tree/master/examples/wtccc_bcell
[`simulation_template.m`]: https://github.com/SUwonglab/rss-net/tree/master/examples/wtccc_bcell/simulation_template.m
[`run_simulation.m`]: https://github.com/SUwonglab/rss-net/tree/master/examples/wtccc_bcell/run_simulation.m
[`run_simulation.sbatch`]: https://github.com/SUwonglab/rss-net/tree/master/examples/wtccc_bcell/run_simulation.sbatch
[structure array]: https://www.mathworks.com/help/matlab/ref/struct.html
[`rss_net.m`]: https://github.com/SUwonglab/rss-net/blob/master/src/rss_net.m
[`calc_bf.m`]: https://github.com/SUwonglab/rss-net/blob/master/src/calc_bf.m
[`compute_pip.m`]: https://github.com/stephenslab/rss/blob/master/src_vb/compute_pip.m
[zenodo-wtcccbcell]: https://doi.org/10.5281/zenodo.3700214 

## Overview

Here we describe an end-to-end RSS-NET analysis of two synthetic datasets
that are simulated from real genotypes of 348965 genome-wide SNPs from
1458 individuals in the UK Blood Service Control Group
([Wellcome Trust Case Control Consortium, 2007](https://doi.org/10.1038/nature05911))
and a gene regulatory network inferred for B cells
[(ENCODE Project Consortium, 2012)](https://doi.org/10.1038/nature11247).
This example gives a quick view of how RSS-NET works.

To reproduce results of this example,
please use scripts in the directory [`wtccc_bcell/`][],
and follow the step-by-step guide below.
Before running any script in [`wtccc_bcell/`][],
please [install](setup.html) RSS-NET.

With the software installed and the input data downloaded,
one should be able to run this example by simply typing the following line in shell:

```{r, eval=FALSE, engine='zsh'}
$ sbatch run_simulation.sbatch
```

If a different directory is used to store the software, input data
and/or output data, please modify file paths in the given scripts accordingly.

## Step-by-step illustration

### Download data files

All data files required to run this example are freely available at Zenodo
[![DOI](https://zenodo.org/badge/DOI/10.5281/zenodo.3700214.svg)][zenodo-wtcccbcell].
Please contact me (`xiangzhu[at]stanford.edu`)
if you have any trouble accessing these files.
After a complete download, you should see the following files.

```{r, eval=FALSE, engine='zsh'}
$ tree .
.
├── m0_data.mat
├── m0_true_gene.mat
├── m1_data.mat
├── m1_true_gene.mat
├── Primary_B_cells_from_peripheral_blood_gene2gene.mat
├── wtccc_bcell_data.md5
├── wtccc_bcell_results.zip
├── WTCCC_Primary_B_cells_from_peripheral_blood_snp2gene_cis.mat
├── WTCCC_Primary_B_cells_from_peripheral_blood_snp2net.mat
├── WTCCC_snp2gene.mat
└── wtccc_tidy_snp2gene_ncbi35togrch37.mat

0 directories, 11 files
```

To help readers confirm if they use the same files as we do,
we report 128-bit MD5 hashes of all files in `wtccc_bcell_data.md5`.

To help readers confirm if they can reproduce results of this example,
we also provide the results (`wtccc_bcell_results.zip`) in the same Zenodo deposit.

For simplicity and generality,
we introduce the following short-hand notations.

```matlab
gwas = 'WTCCC';
net  = 'Primary_B_cells_from_peripheral_blood'; 
```

#### 1. `m*_data.mat`: simulated GWAS summary statistics and reference LD estimates

Here we simulate two synthetic datasets from WTCCC genotypes and B cell network:
one (`m0_data.mat`) based on RSS-NET baseline model $M_0$,
and the other (`m1_data.mat`) based on RSS-NET enrichment model $M_1$.
We match `m*_data.mat` by the number of trait-associated SNPs and
the proportion of phenotypic variation explained by all SNPs.
By this design, `m*_data.mat` have the same amount of genetic signals,
and they only differ in how genetic signals are distributed:
for the enrichment dataset `m1_data.mat`
genetic associations are enriched in the B cell network;
for the baseline dataset `m0_data.mat`
genetic associations are randomly distributed across the genome.

```{r, eval=FALSE, engine='zsh'}
$ md5sum m*_data.mat
5d652796b4cc358843920b1edf2819fc  m0_data.mat
996eae1410f89607f1debefb31ec7811  m1_data.mat

$ du -sh m*_data.mat
3.0G	m0_data.mat
3.0G	m1_data.mat
```

Let's look at the contents of `m*_data.mat`.

```matlab
>> sumstat=matfile('m0_data.mat'); 
>> sumstat
  matlab.io.MatFile
  Properties:
      Properties.Source: 'm0_data.mat'
    Properties.Writable: false                                                  
                  SiRiS: [22x1 cell]                                            
                betahat: [22x1 cell]                                            
                    chr: [22x1 cell]                                            
                    pos: [22x1 cell]                                            
                     se: [22x1 cell]
```

GWAS summary statistics and LD estimates are stored as
[cell arrays](https://www.mathworks.com/help/matlab/cell-arrays.html).
RSS-NET only uses the following variables:

- `betahat{j,1}`, single-SNP effect size estimates of all SNPs on chromosome `j`;
- `se{j,1}`, standard errors of `betahat{j, 1}`;
- `chr{j,1}` and `pos{j, 1}`, physical positions of these SNPs (GRCh37 build);
- `SiRiS{j,1}`, a [sparse matrix](https://www.mathworks.com/help/matlab/sparse-matrices.html),
defined as `repmat((1./se),1,p) .* R .* repmat((1./se)',p,1)`,
where `R` is the estimated LD matrix of these `p` SNPs.

#### 2. `${gwas}_snp2gene.mat`: physical distance between SNPs and genes

This file contains the physical distance between
each GWAS SNP and each protein-coding gene, within 1 Mb.
This file corresponds to ${\bf G}_j$ in the RSS-NET model.

```{r, eval=FALSE, engine='zsh'}
$ md5sum WTCCC_snp2gene.mat 
50da1887601df3a0914380e2ea9b1be7  WTCCC_snp2gene.mat

$ du -sh WTCCC_snp2gene.mat
645M	WTCCC_snp2gene.mat
```

In this example, there are 18334 genes and 348965 SNPs.

```matlab
>> snp2gene=matfile('WTCCC_snp2gene.mat');
>> snp2gene
  matlab.io.MatFile
  Properties:
      Properties.Source: 'WTCCC_snp2gene.mat'
    Properties.Writable: false                                                         
                    chr: [348965x1   double]                                           
                  colid: [41897736x1 int32]                                            
             ncbi35_pos: [348965x1   int32]                                            
                numgene: [1x1        int32]                                            
                 numsnp: [1x1        int32]                                            
                    pos: [348965x1   int32]                                            
                  rowid: [41897736x1 int32]                                            
                    val: [41897736x1 double]                                           

>> [snp2gene.numgene snp2gene.numsnp]
    18334   348965
```

The SNP-to-gene distance information is captured
by a three-column matrix `[colid rowid val]`.
For example, the distance between gene `7` and SNP `161` is `2855196` bps. 

```matlab
>> colid=snp2gene.colid; rowid=snp2gene.rowid; val=snp2gene.val;
>> [colid(6688) rowid(6688) val(6688)]
         7       161   2855196
```

#### 3. `${net}_gene2gene.mat`: gene regulatory network

This file contains information of gene-to-gene
connections in a given regulatory network.

```{r, eval=FALSE, engine='zsh'}
$ md5sum Primary_B_cells_from_peripheral_blood_gene2gene.mat
0098dd6d85762b18d5d594da6c3cfe93  Primary_B_cells_from_peripheral_blood_gene2gene.mat

$ du -sh Primary_B_cells_from_peripheral_blood_gene2gene.mat
1.8M	Primary_B_cells_from_peripheral_blood_gene2gene.mat
```

```matlab
>> gene2gene=matfile('Primary_B_cells_from_peripheral_blood_gene2gene.mat');
>> gene2gene
  matlab.io.MatFile
  Properties:
      Properties.Source: 'Primary_B_cells_from_peripheral_blood_gene2gene.mat'
    Properties.Writable: false                                                                                          
                  colid: [110062x1 int32]                                                                               
                numgene: [1x1      int32]                                                                               
                  rowid: [110062x1 int32]                                                                               
                    val: [110062x1 double]
```

For implementation convenience, this file contains the trivial case
where each gene is mapped to itself with `val=1`.

```matlab
>> colid=gene2gene.colid; rowid=gene2gene.rowid; val=gene2gene.val;             
>> [gene2gene.numgene sum(colid==rowid) unique(val(colid==rowid))]
   18334   18334       1
```

For a given network, transcription factors (TFs) are stored in `rowid`
and target genes (TGs) are stored in `colid`.
In this example there are 3018 TGs and 436 TFs.
Among these TFs and TGs, there are 91728 edges.
The edge weights range from 0.61 to 1.
These TF-to-TG connections and edge weights correspond to
$\{{\bf T}_g,v_{gt}\}$ in the RSS-NET model.

```matlab
>> [length(unique(colid(colid ~= rowid))) length(unique(rowid(colid ~= rowid)))]
        3018         436

>> [length(colid(colid ~= rowid)) length(rowid(colid ~= rowid))]
       91728       91728

>> val_tftg = val(colid ~= rowid);
>> [min(val_tftg) quantile(val_tftg, 0.25) median(val_tftg) quantile(val_tftg, 0.75) max(val_tftg)]
    0.6101    0.6278    0.6515    0.6885    1.0000      
```

#### 4. `${gwas}_${net}_snp2net.mat`: SNP-to-network proximity annotation

This file contains binary annotations
whether a SNP is "near" the given network,
that is, within 100 kb of any network element
(TF, TG or associated regulatory elements).

```{r, eval=FALSE, engine='zsh'}
$ md5sum WTCCC_Primary_B_cells_from_peripheral_blood_snp2net.mat
701264f811c0f5edd582ee3eec5cfd15  WTCCC_Primary_B_cells_from_peripheral_blood_snp2net.mat

$ du -sh WTCCC_Primary_B_cells_from_peripheral_blood_snp2net.mat
9.4M	WTCCC_Primary_B_cells_from_peripheral_blood_snp2net.mat
```

In this example, there are 348965 GWAS SNPs and
121308 of them are near the B cell network (i.e. `val=1`).

```matlab
>> snp2net=matfile('WTCCC_Primary_B_cells_from_peripheral_blood_snp2net.mat');
>> snp2net
  matlab.io.MatFile
  Properties:
      Properties.Source: 'WTCCC_Primary_B_cells_from_peripheral_blood_snp2net.mat'
    Properties.Writable: false                                                                                              
                    chr: [348965x1 double]                                                                                  
             ncbi35_pos: [348965x1 int32]                                                                                   
                    pos: [348965x1 int32]                                                                                   
                  snpid: [348965x1 int32]                                                                                   
                    val: [348965x1 double]                                                                                  
                 window: [1x1      double]                                                                                  

>> [length(snp2net.val) sum(snp2net.val) snp2net.window]
      348965      121308      100000

>> unique(snp2net.val)'
     0     1
```

#### 5. `${gwas}_${net}_snp2gene_cis.mat`: SNP-to-gene cis regulation

This file contains the SNP-to-gene cis regulation scores.
(For simplicity we do not use context-matching cis eQTL studies in our simulations.)
This file corresponds to $(c_{jg}-1)$ in the RSS-NET model.

```{r, eval=FALSE, engine='zsh'}
$ md5sum WTCCC_Primary_B_cells_from_peripheral_blood_snp2gene_cis.mat
b77eebd576eee6d19ac3c8ed94681f00  WTCCC_Primary_B_cells_from_peripheral_blood_snp2gene_cis.mat

$ du -sh WTCCC_Primary_B_cells_from_peripheral_blood_snp2gene_cis.mat
4.5M	WTCCC_Primary_B_cells_from_peripheral_blood_snp2gene_cis.mat
```

```matlab
>> snp2gene_cis = matfile('WTCCC_Primary_B_cells_from_peripheral_blood_snp2gene_cis.mat');
>> snp2gene_cis
  matlab.io.MatFile
  Properties:
      Properties.Source: 'WTCCC_Primary_B_cells_from_peripheral_blood_snp2gene_cis.mat'
    Properties.Writable: false                                                                                                   
                  colid: [291181x1 int32]                                                                                        
                     l0: [1x1      double]                                                                                       
                     l1: [1x1      double]                                                                                       
                numgene: [1x1      int32]                                                                                        
                 numsnp: [1x1      int32]                                                                                        
                  rowid: [291181x1 int32]                                                                                        
                    val: [291181x1 double]                                                                                       
                 window: [1x1      double]
```

In this example, there are 291181 SNP-gene pairs
with cis regulation scores available,
consisting of 111097 SNPs and 5345 genes.
The cis regulation scores (`val`) range from 0.02 to 0.90.

```matlab
>> [snp2gene_cis.numsnp length(unique(snp2gene_cis.rowid)) snp2gene_cis.numgene length(unique(snp2gene_cis.colid))]
   348965   111097    18334     5345

>> val=snp2gene_cis.val;
>> [min(val) quantile(val, 0.25) median(val) quantile(val, 0.75) max(val)]
    0.0202    0.1615    0.2661    0.3864    0.8963
```

### Run RSS-NET analysis

To facilitate running RSS-NET on real data,
we provide a generic script [`simulation_template.m`][].
For the present example, the RSS-NET analysis of two synthetic datasets
is implemented by [`run_simulation.m`][] and [`run_simulation.sbatch`][].

Please see scripts in [`wtccc_bcell/`][] for full details.
Below we only highlight the important parts.

#### 1. Specify input data files

First we need to tell RSS-NET where to find all required input files,
by creating the following [structure array][] `data`.
Please ensure that you save all downloaded files in `dat_path`.

```matlab
data.sumstats_file     = strcat(dat_path,trial_name,'_data.mat');
data.snp2net_file      = strcat(dat_path,gwas_name,'_',net_name,'_snp2net.mat');
data.snp2gene_file     = strcat(dat_path,gwas_name,'_snp2gene.mat');
data.gene2gene_file    = strcat(dat_path,net_name,'_gene2gene.mat');
data.snp2gene_cis_file = strcat(dat_path,gwas_name,'_',net_name,'_snp2gene_cis.mat');
```

#### 2. Specify a hyper-parameter grid

Next we need feed a hyper-parameter grid to RSS-NET,
by creating the following [structure array][] `hyper`.
The total computational cost of RSS-NET is proportional to the grid size.
Hence, if you want to finish running this example faster,
you can always reduce the grid size.

```matlab
hyper.theta0 = true_theta0;
hyper.sigma0 = true_sigma0;
hyper.theta  = [0 ((true_theta-0.5):0.25:(true_theta+0.5))];
hyper.sigma  = [0 ((true_sigma-0.5):0.25:(true_sigma+0.5))];
```

#### 3. Initialize variational parameters

We can provide initial values for variational parameters in RSS-NET as follows.
If we do not provide initial values,
RSS-NET will use a random initialization.

```matlab
options.alpha = alpha0;
options.mu    = mu0;
```

#### 4. Fit the RSS-NET model

We use [`rss_net.m`][] to fit the RSS-NET model on a given dataset.
The model fitting results consist of variational lower bounds `logw` and
variational parameter estimates `[alpha mu s]` for the given hyper-parameter grid.

```matlab
[logw,alpha,mu,s,param] = rss_net(data,hyper,options);
```

#### 5. Summarize network-level enrichments

To assess whether a regulatory network is enriched
for genetic associations with a trait,
we evaluate a Bayes factor (BF) comparing
the baseline model ($M_0:\theta=0~\text{and}~\sigma^2=0$)
in RSS-NET with the enrichment model ($M_1:\theta>0~\text{or}~\sigma^2>0$).
The BF computation is implemented in [`calc_bf.m`][].

```matlab
log10_bf = calc_bf(logw,param);
```

#### 6. Summarize gene-level associations

To summarize association between a locus and a trait,
we compute $P_1^{\sf net}$, the posterior probability that at least
one SNP $j$ in the locus is associated with the trait ($\beta_j\neq 0$):
$$
P_1^{\sf net}=1-\Pr(\beta_j=0,~\forall j\in\text{locus}~|~\text{data},M_1).
$$
As in [Zhu et al (2020)][], a locus is defined as the transcribed region
of a gene plus 100 kb upstream and downstream.
The locus definition is provided in `wtccc_tidy_snp2gene_ncbi35togrch37.mat`.
The $P_1^{\sf net}$ computation is implemented in [`compute_pip.m`][].

```matlab
seg_path = strcat(dat_path,'wtccc_tidy_snp2gene_ncbi35togrch37.mat');
p1_net   = compute_pip(seg_path,logw,alpha);
```

#### 7. Submit jobs

After understanding the basic workflow of RSS-NET,
we submit two jobs of running RSS-NET on the synthetic datasets `m*_data.mat`.
To this end, we write a simple sbatch script [`run_simulation.sbatch`][],
and submit it to a cluster with [`Slurm`](https://slurm.schedmd.com/) available.

```{r, eval=FALSE, engine='zsh'}
$ squeue -u xiangzhu
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
        66258089_0   broadwl run_simu xiangzhu  R    2:48:17      1 midway2-0009
        66258089_1   broadwl run_simu xiangzhu  R    2:48:17      1 midway2-0011
```

For each job, we request 1 node with 8 CPUs and 32 Gb total memory.
The memory utilized by these two jobs is around 16 Gb each.
The running time of these two jobs is around 4 hours each.

#### 8. Understand analysis results

Running [`run_simulation.sbatch`][] yields two MAT-files `m*_results.mat`.

```matlab
>> m0_res=matfile('m0_results.mat');
>> m1_res=matfile('m1_results.mat');
```

As shown below, RSS-NET yields a small BF ($0.13$) for the baseline dataset `m0_data.mat`
and a large BF ($5.85\times 10^{55}$) for the enrichment dataset `m1_data.mat`. 

```matlab
>> [m0_res.log10_bf m1_res.log10_bf]
   -0.8917   55.7671
```

We also compare gene-level associations inferred from
RSS-NET (`m*_res.pp`) with ground truth (`m*_true_gene.mat`).
RSS-NET shows a higher statistical power in identifying gene-level associations
on the enrichment dataset `m1_data.mat` ($\text{AUC}=0.78$)
than the baseline dataset `m0_data.mat` ($\text{AUC}=0.60$).
This is because RSS-NET automatically exploits the underlying network
enrichment in `m1_data.mat` when assessing genetic associations.

```matlab
>> m1_gene=matfile('m1_true_gene.mat');              
>> m1_mdl = fitglm(m1_res.pp,m1_gene.true_gene,'Distribution','binomial','Link','logit');
>> [X,Y,T,m1_auc] = perfcurve(m1_gene.true_gene,m1_mdl.Fitted.Probability,1);
>> m1_auc
    0.7767

>> m0_gene=matfile('m0_true_gene.mat');                                                  
>> m0_mdl = fitglm(m0_res.pp,m0_gene.true_gene,'Distribution','binomial','Link','logit');
>> [X,Y,T,m0_auc] = perfcurve(m0_gene.true_gene,m0_mdl.Fitted.Probability,1);
>> m0_auc
    0.5954
```
