<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Xiang Zhu" />

<meta name="date" content="2019-08-19" />

<title>RSS-NET analysis of IBD GWAS summary statistics and NK cell regulatory network</title>

<script src="site_libs/header-attrs-2.1/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/sandstone.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">rss-net</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="setup.html">Setup</a>
</li>
<li>
  <a href="example.html">Examples</a>
</li>
<li>
  <a href="function.html">Functions</a>
</li>
<li>
  <a href="faq.html">FAQ</a>
</li>
<li>
  <a href="https://stephenslab.github.io/rss/News.html">News</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/SUwonglab/rss-net">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<div class="fluid-row" id="header">



<h1 class="title toc-ignore">RSS-NET analysis of IBD GWAS summary statistics and NK cell regulatory network</h1>
<h4 class="author">Xiang Zhu</h4>
<h4 class="date">2019-08-19</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-03-03
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>rss-net/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190823code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190823)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190823code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190823)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomSUwonglabrssnettree6520dbc09be84af5b1f4a9d9ca38f95a50650970targetblank6520dbca"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/SUwonglab/rss-net/tree/6520dbc09be84af5b1f4a9d9ca38f95a50650970" target="_blank">6520dbc</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomSUwonglabrssnettree6520dbc09be84af5b1f4a9d9ca38f95a50650970targetblank6520dbca" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/SUwonglab/rss-net/blob/6520dbc09be84af5b1f4a9d9ca38f95a50650970/rmd/ibd2015_nkcell.Rmd" target="_blank">6520dbc</a>
</td>
<td>
Xiang Zhu
</td>
<td>
2020-03-03
</td>
<td>
wflow_publish(“rmd/ibd2015_nkcell.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/SUwonglab/rss-net/be857e069dc51bc5b7779dccb04c93d11525339a/docs/ibd2015_nkcell.html" target="_blank">be857e0</a>
</td>
<td>
Xiang Zhu
</td>
<td>
2020-03-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/SUwonglab/rss-net/blob/69e47725c025a6a5e226aa986c15246e24c13b90/rmd/ibd2015_nkcell.Rmd" target="_blank">69e4772</a>
</td>
<td>
Xiang Zhu
</td>
<td>
2020-03-03
</td>
<td>
wflow_publish(“rmd/ibd2015_nkcell.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/SUwonglab/rss-net/5974d280da0308a0da739741c17bc3ecb45562fe/docs/ibd2015_nkcell.html" target="_blank">5974d28</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/SUwonglab/rss-net/blob/4a13f45807786aacfa63dc4ed0ca2e2067abcf76/rmd/ibd2015_nkcell.Rmd" target="_blank">4a13f45</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-03
</td>
<td>
wflow_publish(“rmd/ibd2015_nkcell.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/SUwonglab/rss-net/bf742e05134d16bf1f49e590bac893103bc7d473/docs/ibd2015_nkcell.html" target="_blank">bf742e0</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/SUwonglab/rss-net/blob/a16d453c7ab14ff07600ea8da1ec63c538d151b2/rmd/ibd2015_nkcell.Rmd" target="_blank">a16d453</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-02
</td>
<td>
wflow_publish(“rmd/ibd2015_nkcell.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/SUwonglab/rss-net/785a4239838914d0a47318f9190c42c4b6275ae2/docs/ibd2015_nkcell.html" target="_blank">785a423</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/SUwonglab/rss-net/blob/bda738237a2d508843d9cf3bf62634dc16b8c024/rmd/ibd2015_nkcell.Rmd" target="_blank">bda7382</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-01
</td>
<td>
wflow_publish(“rmd/ibd2015_nkcell.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/SUwonglab/rss-net/830b5816461415255fa4d8205853b11f6b0428e8/docs/ibd2015_nkcell.html" target="_blank">830b581</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/SUwonglab/rss-net/blob/a5b592e1a0fa19749acf3068b8666cfddbb735a9/rmd/ibd2015_nkcell.Rmd" target="_blank">a5b592e</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-01
</td>
<td>
wflow_publish(“rmd/ibd2015_nkcell.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/SUwonglab/rss-net/c3ca100647a4b76c9e74520dd555d3928f37e945/docs/ibd2015_nkcell.html" target="_blank">c3ca100</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-02-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/SUwonglab/rss-net/blob/2e5247ba3de7f13d0a9c0d2e4d2ee103cfd1fe07/rmd/ibd2015_nkcell.Rmd" target="_blank">2e5247b</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-02-29
</td>
<td>
wflow_publish(“rmd/ibd2015_nkcell.Rmd”)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="overview" class="section level2">
<h2>Overview</h2>
<p>Here we describe an end-to-end RSS-NET analysis of inflammatory bowel disease (IBD) GWAS summary statistics <a href="https://www.ncbi.nlm.nih.gov/pubmed/26192919">(Liu et al, 2015)</a> and a gene regulatory network inferred for natural killer (NK) cells. This example illustrates the actual data analyses performed in <a href="xxx">Zhu et al (2019)</a>.</p>
<p>To reproduce results of this example, please use scripts in the directory <a href="xxx"><code>script_dir</code></a>, and follow the step-by-step guide below. Before running any script in <a href="xxx"><code>script_dir</code></a>, please <a href="setup.html">install</a> RSS-NET.</p>
<p>Since a real genome-wide analysis is conducted here, this example is more complicated than the previous <a href="xxx">simulation example</a>. It is advisable to go through the previous <a href="xxx">simulation example</a> before diving into this real data example.</p>
<p>Note that the working directory here is assumed to be <code>wdtba</code>. Please modify scripts accordingly if a different directory is used.</p>
</div>
<div id="step-by-step-illustration" class="section level2">
<h2>Step-by-step illustration</h2>
<div id="download-input-data-files" class="section level3">
<h3>Download input data files</h3>
<div id="ibd2015_sumstat.mat-processed-gwas-summary-statistics-and-ld-matrix-estimates" class="section level4">
<h4>1. <code>ibd2015_sumstat.mat</code>: processed GWAS summary statistics and LD matrix estimates</h4>
<p>This file is large (43G) because it has a LD matrix of 1.1 million common SNPs. Please contact me (<code>xiangzhu[at]stanford.edu</code>) if you have trouble accessing this file.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode zsh"><code class="sourceCode zsh"><span id="cb1-1"><a href="#cb1-1"></a>$ md5sum ibd2015_sumstat.mat</span>
<span id="cb1-2"><a href="#cb1-2"></a>ad1763079ee7e46b21722f74e037a230  ibd2015_sumstat.mat</span>
<span id="cb1-3"><a href="#cb1-3"></a></span>
<span id="cb1-4"><a href="#cb1-4"></a>$ <span class="kw">du</span> -sh ibd2015_sumstat.mat                                             </span>
<span id="cb1-5"><a href="#cb1-5"></a>43G ibd2015_sumstat.mat</span></code></pre></div>
<p>Let’s look at the contents of <code>ibd2015_sumstat.mat</code>.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode matlab"><code class="sourceCode matlab"><span id="cb2-1"><a href="#cb2-1"></a>&gt;&gt; sumstat = matfile(<span class="st">&#39;ibd2015_sumstat.mat&#39;</span>);</span>
<span id="cb2-2"><a href="#cb2-2"></a>&gt;&gt; sumstat</span>
<span id="cb2-3"><a href="#cb2-3"></a>  matlab.io.MatFile</span>
<span id="cb2-4"><a href="#cb2-4"></a>  Properties:</span>
<span id="cb2-5"><a href="#cb2-5"></a>      Properties.Source: <span class="st">&#39;ibd2015_sumstat.mat&#39;</span></span>
<span id="cb2-6"><a href="#cb2-6"></a>    Properties.Writable: false                                                          </span>
<span id="cb2-7"><a href="#cb2-7"></a>                     BR: [<span class="fl">22</span>x1 cell]                                                    </span>
<span id="cb2-8"><a href="#cb2-8"></a>                      R: [<span class="fl">22</span>x1 cell]                                                    </span>
<span id="cb2-9"><a href="#cb2-9"></a>                  SiRiS: [<span class="fl">22</span>x1 cell]                                                    </span>
<span id="cb2-10"><a href="#cb2-10"></a>                betahat: [<span class="fl">22</span>x1 cell]                                                    </span>
<span id="cb2-11"><a href="#cb2-11"></a>                    chr: [<span class="fl">22</span>x1 cell]                                                    </span>
<span id="cb2-12"><a href="#cb2-12"></a>                    pos: [<span class="fl">22</span>x1 cell]                                                    </span>
<span id="cb2-13"><a href="#cb2-13"></a>                     se: [<span class="fl">22</span>x1 cell]</span></code></pre></div>
<p>GWAS summary statistics and LD estimates are stored as <a href="https://www.mathworks.com/help/matlab/cell-arrays.html">cell arrays</a>. RSS-NET only uses the following variables:</p>
<ul>
<li><code>betahat{j,1}</code>, single-SNP effect size estimates of all SNPs on chromosome <code>j</code>;</li>
<li><code>se{j,1}</code>, standard errors of <code>betahat{j, 1}</code>;</li>
<li><code>chr{j,1}</code> and <code>pos{j, 1}</code>, physical positions of these SNPs (GRCh37 build);</li>
<li><code>SiRiS{j,1}</code>, a <a href="https://www.mathworks.com/help/matlab/sparse-matrices.html">sparse matrix</a>, defined as <code>repmat((1./se),1,p) .* R .* repmat((1./se)',p,1)</code>, where <code>R</code> is the estimated LD matrix of these <code>p</code> SNPs.</li>
</ul>
</div>
<div id="ibd2015_snp2gene.mat-physical-distance-between-snps-and-genes" class="section level4">
<h4>2. <code>ibd2015_snp2gene.mat</code>: physical distance between SNPs and genes</h4>
<p>This file contains the physical distance between each GWAS SNP and each protein-coding gene, within 1 Mb. This file corresponds to <span class="math inline">\({\bf G}_j\)</span> in the RSS-NET model.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode zsh"><code class="sourceCode zsh"><span id="cb3-1"><a href="#cb3-1"></a>$ md5sum ibd2015_snp2gene.mat </span>
<span id="cb3-2"><a href="#cb3-2"></a>7832838e2675e4cf3b85f471fed95554  ibd2015_snp2gene.mat</span>
<span id="cb3-3"><a href="#cb3-3"></a></span>
<span id="cb3-4"><a href="#cb3-4"></a>$ <span class="kw">du</span> -sh ibd2015_snp2gene.mat </span>
<span id="cb3-5"><a href="#cb3-5"></a>224M    ibd2015_snp2gene.mat</span></code></pre></div>
<p>In this example, there are 18334 SNPs and 1081481 genes.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode matlab"><code class="sourceCode matlab"><span id="cb4-1"><a href="#cb4-1"></a>&gt;&gt; snp2gene = matfile(<span class="st">&#39;ibd2015_snp2gene.mat&#39;</span>);</span>
<span id="cb4-2"><a href="#cb4-2"></a>&gt;&gt; snp2gene                                   </span>
<span id="cb4-3"><a href="#cb4-3"></a>  matlab.io.MatFile</span>
<span id="cb4-4"><a href="#cb4-4"></a>  Properties:</span>
<span id="cb4-5"><a href="#cb4-5"></a>      Properties.Source: <span class="st">&#39;ibd2015_snp2gene.mat&#39;</span></span>
<span id="cb4-6"><a href="#cb4-6"></a>    Properties.Writable: false                                                                </span>
<span id="cb4-7"><a href="#cb4-7"></a>                    chr: [<span class="fl">1081481</span>x1  int32]                                                   </span>
<span id="cb4-8"><a href="#cb4-8"></a>                  colid: [<span class="fl">14126805</span>x1 int32]                                                   </span>
<span id="cb4-9"><a href="#cb4-9"></a>                numgene: [<span class="fl">1</span>x1        int32]                                                   </span>
<span id="cb4-10"><a href="#cb4-10"></a>                 numsnp: [<span class="fl">1</span>x1        int32]                                                   </span>
<span id="cb4-11"><a href="#cb4-11"></a>                    pos: [<span class="fl">1081481</span>x1  int32]                                                   </span>
<span id="cb4-12"><a href="#cb4-12"></a>                  rowid: [<span class="fl">14126805</span>x1 int32]                                                   </span>
<span id="cb4-13"><a href="#cb4-13"></a>                    val: [<span class="fl">14126805</span>x1 double]</span>
<span id="cb4-14"><a href="#cb4-14"></a></span>
<span id="cb4-15"><a href="#cb4-15"></a>&gt;&gt; [snp2gene.numgene snp2gene.numsnp]                                                              </span>
<span id="cb4-16"><a href="#cb4-16"></a>     <span class="fl">18334</span>   <span class="fl">1081481</span></span></code></pre></div>
<p>The SNP-to-gene distance information is captured by a three-column matrix <code>[colid rowid val]</code>. For example, the distance between gene <code>1</code> and SNP <code>6</code> is <code>978947</code> bps.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode matlab"><code class="sourceCode matlab"><span id="cb5-1"><a href="#cb5-1"></a>&gt;&gt; colid=snp2gene.colid; rowid=snp2gene.rowid; val=snp2gene.val;</span>
<span id="cb5-2"><a href="#cb5-2"></a>&gt;&gt; [colid(<span class="fl">6</span>) rowid(<span class="fl">6</span>) val(<span class="fl">6</span>)]</span>
<span id="cb5-3"><a href="#cb5-3"></a>        <span class="fl">1</span>        <span class="fl">6</span>   <span class="fl">978947</span></span></code></pre></div>
</div>
<div id="primary_natural_killer_cells_from_peripheral_blood_gene2gene.mat-gene-regulatory-network" class="section level4">
<h4>3. <code>Primary_Natural_Killer_cells_from_peripheral_blood_gene2gene.mat</code>: gene regulatory network</h4>
<p>This file contains information of gene-to-gene connections in a given regulatory network.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode zsh"><code class="sourceCode zsh"><span id="cb6-1"><a href="#cb6-1"></a>$ md5sum Primary_Natural_Killer_cells_from_peripheral_blood_gene2gene.mat </span>
<span id="cb6-2"><a href="#cb6-2"></a>35ac724b86f7777d87116cc48166caa2  Primary_Natural_Killer_cells_from_peripheral_blood_gene2gene.mat</span>
<span id="cb6-3"><a href="#cb6-3"></a></span>
<span id="cb6-4"><a href="#cb6-4"></a>$ <span class="kw">du</span> -sh Primary_Natural_Killer_cells_from_peripheral_blood_gene2gene.mat</span>
<span id="cb6-5"><a href="#cb6-5"></a>1.7M    Primary_Natural_Killer_cells_from_peripheral_blood_gene2gene.mat</span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="sourceCode matlab"><code class="sourceCode matlab"><span id="cb7-1"><a href="#cb7-1"></a>&gt;&gt; gene2gene = matfile(<span class="st">&#39;Primary_Natural_Killer_cells_from_peripheral_blood_gene2gene.mat&#39;</span>);</span>
<span id="cb7-2"><a href="#cb7-2"></a>&gt;&gt; gene2gene</span>
<span id="cb7-3"><a href="#cb7-3"></a>  matlab.io.MatFile</span>
<span id="cb7-4"><a href="#cb7-4"></a>  Properties:</span>
<span id="cb7-5"><a href="#cb7-5"></a>      Properties.Source: <span class="st">&#39;Primary_Natural_Killer_cells_from_peripheral_blood_gene2gene.mat&#39;</span></span>
<span id="cb7-6"><a href="#cb7-6"></a>    Properties.Writable: false                                                                                                            </span>
<span id="cb7-7"><a href="#cb7-7"></a>                  colid: [<span class="fl">110733</span>x1 int32]                                                                                                 </span>
<span id="cb7-8"><a href="#cb7-8"></a>                numgene: [<span class="fl">1</span>x1      int32]                                                                                                 </span>
<span id="cb7-9"><a href="#cb7-9"></a>                  rowid: [<span class="fl">110733</span>x1 int32]                                                                                                 </span>
<span id="cb7-10"><a href="#cb7-10"></a>                    val: [<span class="fl">110733</span>x1 double]</span></code></pre></div>
<p>For implementation convenience, this file contains the trivial case where each gene is mapped to itself with <code>val=1</code>.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode matlab"><code class="sourceCode matlab"><span id="cb8-1"><a href="#cb8-1"></a>&gt;&gt; colid=gene2gene.colid; rowid=gene2gene.rowid; val=gene2gene.val;             </span>
<span id="cb8-2"><a href="#cb8-2"></a>&gt;&gt; [gene2gene.numgene sum(colid==rowid) unique(val(colid==rowid))]</span>
<span id="cb8-3"><a href="#cb8-3"></a>   <span class="fl">18334</span>   <span class="fl">18334</span>       <span class="fl">1</span></span></code></pre></div>
<p>For a given network, transcription factors (TFs) are stored in <code>rowid</code> and target genes (TGs) are stored in <code>colid</code>. In this example there are 3105 TGs and 376 TFs. Among these TFs and TGs, there are 92399 edges. The edge weights range from 0.61 to 1. These TF-to-TG connections and edge weights correspond to <span class="math inline">\(\{{\bf T}_g,v_{gt}\}\)</span> in the RSS-NET model.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode matlab"><code class="sourceCode matlab"><span id="cb9-1"><a href="#cb9-1"></a>&gt;&gt; [length(unique(colid(colid ~= rowid))) length(unique(rowid(colid ~= rowid)))]</span>
<span id="cb9-2"><a href="#cb9-2"></a>        <span class="fl">3105</span>         <span class="fl">376</span></span>
<span id="cb9-3"><a href="#cb9-3"></a>&gt;&gt; [length(colid(colid ~= rowid)) length(rowid(colid ~= rowid))]</span>
<span id="cb9-4"><a href="#cb9-4"></a>       <span class="fl">92399</span>       <span class="fl">92399</span></span>
<span id="cb9-5"><a href="#cb9-5"></a></span>
<span id="cb9-6"><a href="#cb9-6"></a>&gt;&gt; val_tftg = val(colid ~= rowid);</span>
<span id="cb9-7"><a href="#cb9-7"></a>&gt;&gt; [min(val_tftg) quantile(val_tftg, <span class="fl">0.25</span>) median(val_tftg) quantile(val_tftg, <span class="fl">0.75</span>) max(val_tftg)]</span>
<span id="cb9-8"><a href="#cb9-8"></a>    <span class="fl">0.6138</span>    <span class="fl">0.6324</span>    <span class="fl">0.6568</span>    <span class="fl">0.6949</span>    <span class="fl">1.0000</span>       </span></code></pre></div>
</div>
<div id="ibd2015_primary_natural_killer_cells_from_peripheral_blood_snp2net.mat-snp-to-network-proximity-annotation" class="section level4">
<h4>4. <code>ibd2015_Primary_Natural_Killer_cells_from_peripheral_blood_snp2net.mat</code>: SNP-to-network proximity annotation</h4>
<p>This file contains binary annotations whether a SNP is “near” the given network, that is, within 100 kb of any network element (TF, TG or associated regulatory elements).</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode zsh"><code class="sourceCode zsh"><span id="cb10-1"><a href="#cb10-1"></a>$ md5sum ibd2015_Primary_Natural_Killer_cells_from_peripheral_blood_snp2net.mat</span>
<span id="cb10-2"><a href="#cb10-2"></a>d96cd9b32759f954cc37680dc6aeafd8  ibd2015_Primary_Natural_Killer_cells_from_peripheral_blood_snp2net.mat</span>
<span id="cb10-3"><a href="#cb10-3"></a></span>
<span id="cb10-4"><a href="#cb10-4"></a>$ <span class="kw">du</span> -sh ibd2015_Primary_Natural_Killer_cells_from_peripheral_blood_snp2net.mat</span>
<span id="cb10-5"><a href="#cb10-5"></a>21M ibd2015_Primary_Natural_Killer_cells_from_peripheral_blood_snp2net.mat</span></code></pre></div>
<p>In this example, there are 1081481 GWAS SNPs and 382443 of them are near the NK cell network (i.e. <code>val=1</code>).</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode matlab"><code class="sourceCode matlab"><span id="cb11-1"><a href="#cb11-1"></a>&gt;&gt; snp2net = matfile(<span class="st">&#39;ibd2015_Primary_Natural_Killer_cells_from_peripheral_blood_snp2net.mat&#39;</span>);</span>
<span id="cb11-2"><a href="#cb11-2"></a>&gt;&gt; snp2net</span>
<span id="cb11-3"><a href="#cb11-3"></a>  matlab.io.MatFile</span>
<span id="cb11-4"><a href="#cb11-4"></a>  Properties:</span>
<span id="cb11-5"><a href="#cb11-5"></a>      Properties.Source: <span class="st">&#39;ibd2015_Primary_Natural_Killer_cells_from_peripheral_blood_snp2net.mat&#39;</span></span>
<span id="cb11-6"><a href="#cb11-6"></a>    Properties.Writable: false</span>
<span id="cb11-7"><a href="#cb11-7"></a>                    chr: [<span class="fl">1081481</span>x1 int32]</span>
<span id="cb11-8"><a href="#cb11-8"></a>                    pos: [<span class="fl">1081481</span>x1 int32]</span>
<span id="cb11-9"><a href="#cb11-9"></a>                  snpid: [<span class="fl">1081481</span>x1 int32]</span>
<span id="cb11-10"><a href="#cb11-10"></a>                    val: [<span class="fl">1081481</span>x1 double]</span>
<span id="cb11-11"><a href="#cb11-11"></a>                 window: [<span class="fl">1</span>x1       double]</span>
<span id="cb11-12"><a href="#cb11-12"></a>                 </span>
<span id="cb11-13"><a href="#cb11-13"></a>&gt;&gt; [length(snp2net.val) sum(snp2net.val) snp2net.window]</span>
<span id="cb11-14"><a href="#cb11-14"></a>     <span class="fl">1081481</span>      <span class="fl">382443</span>      <span class="fl">100000</span></span>
<span id="cb11-15"><a href="#cb11-15"></a>&gt;&gt; unique(snp2net.val)&#39;</span>
<span id="cb11-16"><a href="#cb11-16"></a>     <span class="fl">0</span>     <span class="fl">1</span>     </span></code></pre></div>
</div>
<div id="ibd2015_nk_snp2gene_cis.mat-snp-to-gene-cis-regulation" class="section level4">
<h4>5. <code>ibd2015_NK_snp2gene_cis.mat</code>: SNP-to-gene cis regulation</h4>
<p>This file contains the SNP-to-gene cis regulation scores derived from context-matching cis eQTL studies. This file corresponds to <span class="math inline">\((c_{jg}-1)\)</span> in the RSS-NET model.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode zsh"><code class="sourceCode zsh"><span id="cb12-1"><a href="#cb12-1"></a>$ md5sum ibd2015_NK_snp2gene_cis.mat</span>
<span id="cb12-2"><a href="#cb12-2"></a>dedad8e25773fad69576dbce0f7d9f93  ibd2015_NK_snp2gene_cis.mat</span>
<span id="cb12-3"><a href="#cb12-3"></a></span>
<span id="cb12-4"><a href="#cb12-4"></a>$ <span class="kw">du</span> -sh ibd2015_NK_snp2gene_cis.mat</span>
<span id="cb12-5"><a href="#cb12-5"></a>165M    ibd2015_NK_snp2gene_cis.mat</span></code></pre></div>
<div class="sourceCode" id="cb13"><pre class="sourceCode matlab"><code class="sourceCode matlab"><span id="cb13-1"><a href="#cb13-1"></a>&gt;&gt; snp2gene_cis = matfile(<span class="st">&#39;ibd2015_NK_snp2gene_cis.mat&#39;</span>);</span>
<span id="cb13-2"><a href="#cb13-2"></a>&gt;&gt; snp2gene_cis</span>
<span id="cb13-3"><a href="#cb13-3"></a>  matlab.io.MatFile</span>
<span id="cb13-4"><a href="#cb13-4"></a>  Properties:</span>
<span id="cb13-5"><a href="#cb13-5"></a>      Properties.Source: <span class="st">&#39;ibd2015_NK_snp2gene_cis.mat&#39;</span></span>
<span id="cb13-6"><a href="#cb13-6"></a>    Properties.Writable: false</span>
<span id="cb13-7"><a href="#cb13-7"></a>                  colid: [<span class="fl">10790012</span>x1 int32]</span>
<span id="cb13-8"><a href="#cb13-8"></a>                numgene: [<span class="fl">1</span>x1        int32]</span>
<span id="cb13-9"><a href="#cb13-9"></a>                 numsnp: [<span class="fl">1</span>x1        int32]</span>
<span id="cb13-10"><a href="#cb13-10"></a>                  rowid: [<span class="fl">10790012</span>x1 int32]</span>
<span id="cb13-11"><a href="#cb13-11"></a>                    val: [<span class="fl">10790012</span>x1 double]</span></code></pre></div>
<p>In this example, there are 10790012 SNP-gene pairs with cis regulation scores available, consiting of 829280 SNPs and 18230 genes. The cis regulation scores (<code>val</code>) range from 0 to 0.76.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode matlab"><code class="sourceCode matlab"><span id="cb14-1"><a href="#cb14-1"></a>&gt;&gt; [snp2gene_cis.numsnp length(unique(snp2gene_cis.rowid)) snp2gene_cis.numgene length(unique(snp2gene_cis.colid))]</span>
<span id="cb14-2"><a href="#cb14-2"></a>   <span class="fl">1081481</span>    <span class="fl">829280</span>     <span class="fl">18334</span>     <span class="fl">18230</span></span>
<span id="cb14-3"><a href="#cb14-3"></a>   </span>
<span id="cb14-4"><a href="#cb14-4"></a>&gt;&gt; val=snp2gene_cis.val;</span>
<span id="cb14-5"><a href="#cb14-5"></a>&gt;&gt; [min(val) quantile(val, <span class="fl">0.25</span>) median(val) quantile(val, <span class="fl">0.75</span>) max(val)]</span>
<span id="cb14-6"><a href="#cb14-6"></a>         <span class="fl">0</span>    <span class="fl">0.0226</span>    <span class="fl">0.0636</span>    <span class="fl">0.1172</span>    <span class="fl">0.7622</span>   </span></code></pre></div>
</div>
</div>
<div id="run-rss-net-analysis" class="section level3">
<h3>Run RSS-NET analysis</h3>
<div class="sourceCode" id="cb15"><pre class="sourceCode zsh"><code class="sourceCode zsh"><span id="cb15-1"><a href="#cb15-1"></a>$ <span class="kw">pwd</span></span>
<span id="cb15-2"><a href="#cb15-2"></a>/Users/xiangzhu/GitHub/rss-net/examples/ibd2015_nkcell</span>
<span id="cb15-3"><a href="#cb15-3"></a></span>
<span id="cb15-4"><a href="#cb15-4"></a>$ tree -f</span>
<span id="cb15-5"><a href="#cb15-5"></a><span class="kw">.</span></span>
<span id="cb15-6"><a href="#cb15-6"></a>├── ./analysis_template.m</span>
<span id="cb15-7"><a href="#cb15-7"></a>├── ./ibd2015_nkcell.m</span>
<span id="cb15-8"><a href="#cb15-8"></a>└── ./ibd2015_nkcell.sbatch</span>
<span id="cb15-9"><a href="#cb15-9"></a></span>
<span id="cb15-10"><a href="#cb15-10"></a>0 directories, 3 files</span></code></pre></div>
<div id="submit-job-arrays" class="section level4">
<h4>3. Submit job arrays</h4>
<p>For a given GWAS and a given regulatory network, all RSS-NET analysis tasks are almost identical and they only differs in hyper-parameter values. To exploit this, we run one RSS-NET analysis as a job array with multiple tasks that run in parallel.</p>
<p>To this end, we write a simple sbatch script <a href="xxx"><code>ibd2015_nkcell.sbatch</code></a>, and submit it to a cluster with <a href="https://slurm.schedmd.com/"><code>Slurm</code></a> available.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode zsh"><code class="sourceCode zsh"><span id="cb16-1"><a href="#cb16-1"></a>$ sbatch ibd2015_nkcell.sbatch</span></code></pre></div>
<p>After the submission, multiple jobs should run in different nodes simultaneously.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode zsh"><code class="sourceCode zsh"><span id="cb17-1"><a href="#cb17-1"></a>$ squeue -u xiangzhu</span>
<span id="cb17-2"><a href="#cb17-2"></a>             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST<span class="kw">(</span>REASON<span class="kw">)</span></span>
<span id="cb17-3"><a href="#cb17-3"></a>      62554249_107    owners ibd2015_ xiangzhu  R       0:32      1 sh02-17n12</span>
<span id="cb17-4"><a href="#cb17-4"></a>      62554249_108    owners ibd2015_ xiangzhu  R       0:32      1 sh02-17n12</span>
<span id="cb17-5"><a href="#cb17-5"></a>      62554249_109    owners ibd2015_ xiangzhu  R       0:32      1 sh01-28n08</span>
<span id="cb17-6"><a href="#cb17-6"></a>      62554249_110    owners ibd2015_ xiangzhu  R       0:32      1 sh01-17n18</span>
<span id="cb17-7"><a href="#cb17-7"></a>      62554249_111    owners ibd2015_ xiangzhu  R       0:32      1 sh01-26n33</span>
<span id="cb17-8"><a href="#cb17-8"></a>      62554249_112    owners ibd2015_ xiangzhu  R       0:32      1 sh01-27n30</span></code></pre></div>
<p>Start at Mar 2, 2020, 3:04 PM.</p>
<p>End at Mar 2, 2020, 11:37 PM.</p>
<pre><code>Job ID: 62554249
Array Job ID: 62554249_125
Cluster: sherlock
User/Group: xiangzhu/whwong
State: COMPLETED (exit code 0)
Nodes: 1
Cores per node: 8
CPU Utilized: 1-03:08:31
CPU Efficiency: 74.54% of 1-12:24:40 core-walltime
Job Wall-clock time: 04:33:05
Memory Utilized: 26.58 GB
Memory Efficiency: 85.06% of 31.25 GB</code></pre>
</div>
</div>
</div>
<div id="more-examples" class="section level2">
<h2>More examples</h2>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1"></a>devtools<span class="op">::</span><span class="kw">session_info</span>()</span></code></pre></div>
<pre><code>─ Session info ───────────────────────────────────────────────────────────────
 setting  value                       
 version  R version 3.6.2 (2019-12-12)
 os       macOS Catalina 10.15.3      
 system   x86_64, darwin15.6.0        
 ui       X11                         
 language (EN)                        
 collate  en_US.UTF-8                 
 ctype    en_US.UTF-8                 
 tz       America/Los_Angeles         
 date     2020-03-03                  

─ Packages ───────────────────────────────────────────────────────────────────
 package     * version date       lib source        
 assertthat    0.2.1   2019-03-21 [1] CRAN (R 3.6.0)
 backports     1.1.5   2019-10-02 [1] CRAN (R 3.6.0)
 callr         3.4.2   2020-02-12 [1] CRAN (R 3.6.0)
 cli           2.0.2   2020-02-28 [1] CRAN (R 3.6.0)
 crayon        1.3.4   2017-09-16 [1] CRAN (R 3.6.0)
 desc          1.2.0   2018-05-01 [1] CRAN (R 3.6.0)
 devtools      2.2.2   2020-02-17 [1] CRAN (R 3.6.0)
 digest        0.6.25  2020-02-23 [1] CRAN (R 3.6.0)
 ellipsis      0.3.0   2019-09-20 [1] CRAN (R 3.6.0)
 evaluate      0.14    2019-05-28 [1] CRAN (R 3.6.0)
 fansi         0.4.1   2020-01-08 [1] CRAN (R 3.6.0)
 fs            1.3.1   2019-05-06 [1] CRAN (R 3.6.0)
 git2r         0.26.1  2019-06-29 [1] CRAN (R 3.6.0)
 glue          1.3.1   2019-03-12 [1] CRAN (R 3.6.0)
 htmltools     0.4.0   2019-10-04 [1] CRAN (R 3.6.0)
 httpuv        1.5.2   2019-09-11 [1] CRAN (R 3.6.0)
 knitr         1.28    2020-02-06 [1] CRAN (R 3.6.0)
 later         1.0.0   2019-10-04 [1] CRAN (R 3.6.0)
 magrittr      1.5     2014-11-22 [1] CRAN (R 3.6.0)
 memoise       1.1.0   2017-04-21 [1] CRAN (R 3.6.0)
 pkgbuild      1.0.6   2019-10-09 [1] CRAN (R 3.6.0)
 pkgload       1.0.2   2018-10-29 [1] CRAN (R 3.6.0)
 prettyunits   1.1.1   2020-01-24 [1] CRAN (R 3.6.0)
 processx      3.4.2   2020-02-09 [1] CRAN (R 3.6.0)
 promises      1.1.0   2019-10-04 [1] CRAN (R 3.6.0)
 ps            1.3.2   2020-02-13 [1] CRAN (R 3.6.0)
 R6            2.4.1   2019-11-12 [1] CRAN (R 3.6.1)
 Rcpp          1.0.3   2019-11-08 [1] CRAN (R 3.6.0)
 remotes       2.1.1   2020-02-15 [1] CRAN (R 3.6.0)
 rlang         0.4.5   2020-03-01 [1] CRAN (R 3.6.0)
 rmarkdown     2.1     2020-01-20 [1] CRAN (R 3.6.0)
 rprojroot     1.3-2   2018-01-03 [1] CRAN (R 3.6.0)
 sessioninfo   1.1.1   2018-11-05 [1] CRAN (R 3.6.0)
 stringi       1.4.6   2020-02-17 [1] CRAN (R 3.6.0)
 stringr       1.4.0   2019-02-10 [1] CRAN (R 3.6.0)
 testthat      2.3.2   2020-03-02 [1] CRAN (R 3.6.2)
 usethis       1.5.1   2019-07-04 [1] CRAN (R 3.6.0)
 whisker       0.4     2019-08-28 [1] CRAN (R 3.6.0)
 withr         2.1.2   2018-03-15 [1] CRAN (R 3.6.0)
 workflowr   * 1.6.0   2019-12-19 [1] CRAN (R 3.6.0)
 xfun          0.12    2020-01-13 [1] CRAN (R 3.6.0)
 yaml          2.2.1   2020-02-01 [1] CRAN (R 3.6.0)

[1] /Library/Frameworks/R.framework/Versions/3.6/Resources/library</code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
