<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Xiang Zhu" />

<meta name="date" content="2019-08-19" />

<title>RSS-NET analysis of IBD GWAS summary statistics and NK cell regulatory network</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/sandstone.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(title);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">rss-net</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="setup.html">Setup</a>
</li>
<li>
  <a href="function.html">Functions</a>
</li>
<li>
  <a href="faq.html">FAQ</a>
</li>
<li>
  <a href="https://stephenslab.github.io/rss/News.html">News</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/SUwonglab/rss-net">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">RSS-NET analysis of IBD GWAS summary statistics and NK cell regulatory network</h1>
<h4 class="author">Xiang Zhu</h4>
<h4 class="date">2019-08-19</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-03-09
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>rss-net/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190823code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190823)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190823code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190823)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomSUwonglabrssnettree820a50862044fd4aee5af66b8e935bdf771467d9targetblank820a508a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/SUwonglab/rss-net/tree/820a50862044fd4aee5af66b8e935bdf771467d9" target="_blank">820a508</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomSUwonglabrssnettree820a50862044fd4aee5af66b8e935bdf771467d9targetblank820a508a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    examples/ibd2015_nkcell/ibd2015_Primary_Natural_Killer_cells_from_peripheral_blood_NK_out_66.mat
    Ignored:    examples/ibd2015_nkcell/randinit_results/
    Ignored:    examples/ibd2015_nkcell/results/
    Ignored:    examples/wtccc_bcell/results/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/SUwonglab/rss-net/blob/820a50862044fd4aee5af66b8e935bdf771467d9/rmd/ibd2015_nkcell.Rmd" target="_blank">820a508</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-09
</td>
<td>
wflow_publish(“rmd/ibd2015_nkcell.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/SUwonglab/rss-net/a4241718442f47362b6581697b185afd07f2a850/docs/ibd2015_nkcell.html" target="_blank">a424171</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/SUwonglab/rss-net/5c5e8e118c10fae0d8f22664a7dc383a24fe3406/docs/ibd2015_nkcell.html" target="_blank">5c5e8e1</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/SUwonglab/rss-net/3af1c6c378db1bec0e8ddb8ed82e04dfc62203ca/docs/ibd2015_nkcell.html" target="_blank">3af1c6c</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/SUwonglab/rss-net/blob/9dbe6b7bb828d65c09739f1c8e2fc1a01abcfa5f/rmd/ibd2015_nkcell.Rmd" target="_blank">9dbe6b7</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-06
</td>
<td>
wflow_publish(“rmd/ibd2015_nkcell.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/SUwonglab/rss-net/8d17f949197974ae29d10b8ae5c70f1e275a3e23/docs/ibd2015_nkcell.html" target="_blank">8d17f94</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/SUwonglab/rss-net/blob/2d37013c96be5c175943b2e8b2a66cfa52a1350e/rmd/ibd2015_nkcell.Rmd" target="_blank">2d37013</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-06
</td>
<td>
wflow_publish(“rmd/ibd2015_nkcell.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/SUwonglab/rss-net/7e923a9a1e16f1708334b5a2f7bae2913031e52c/docs/ibd2015_nkcell.html" target="_blank">7e923a9</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/SUwonglab/rss-net/blob/6bf894e2061a84af077cb6568f749232561c9e39/rmd/ibd2015_nkcell.Rmd" target="_blank">6bf894e</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-06
</td>
<td>
wflow_publish(“rmd/ibd2015_nkcell.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/SUwonglab/rss-net/36d2ce3a4a05ae0ecfa56cd05eeb225d78d92b20/docs/ibd2015_nkcell.html" target="_blank">36d2ce3</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/SUwonglab/rss-net/blob/bcc6e4a7dcb89c68af79f5e6cb8d44125b19ab62/rmd/ibd2015_nkcell.Rmd" target="_blank">bcc6e4a</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-06
</td>
<td>
wflow_publish(“rmd/ibd2015_nkcell.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/SUwonglab/rss-net/ada5d78c92da1b4a51fdcc666660053b3bb62785/docs/ibd2015_nkcell.html" target="_blank">ada5d78</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-05
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/SUwonglab/rss-net/blob/1753ab58fb46825949eda7f6744756d94109d9fe/rmd/ibd2015_nkcell.Rmd" target="_blank">1753ab5</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-05
</td>
<td>
wflow_publish(“rmd/ibd2015_nkcell.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/SUwonglab/rss-net/4310e9f463332be6e9d7020d10fdbf4a1f5c712a/docs/ibd2015_nkcell.html" target="_blank">4310e9f</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-05
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/SUwonglab/rss-net/blob/793e39e48f5b4d42d3ec610a59695c3a37201b42/rmd/ibd2015_nkcell.Rmd" target="_blank">793e39e</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-05
</td>
<td>
wflow_publish(“rmd/ibd2015_nkcell.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/SUwonglab/rss-net/269dbf17edbd2889bca74d4644bb24ab9b598a13/docs/ibd2015_nkcell.html" target="_blank">269dbf1</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-05
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/SUwonglab/rss-net/blob/dccb92cf2a834ca2e940dc44f01cb774cbc39747/rmd/ibd2015_nkcell.Rmd" target="_blank">dccb92c</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-05
</td>
<td>
wflow_publish(“rmd/ibd2015_nkcell.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/SUwonglab/rss-net/34ece5fe510e4b2e04e96d28331ba0a5f7f678b5/docs/ibd2015_nkcell.html" target="_blank">34ece5f</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-05
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/SUwonglab/rss-net/blob/de5a209a474ca927685d75d99d601d8277f8477a/rmd/ibd2015_nkcell.Rmd" target="_blank">de5a209</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-05
</td>
<td>
wflow_publish(“rmd/ibd2015_nkcell.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/SUwonglab/rss-net/19bd6c5f4389d8f5d923bcd5de1aa72823ca9ff0/docs/ibd2015_nkcell.html" target="_blank">19bd6c5</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/SUwonglab/rss-net/blob/b464c143b4c1dd1508ce1656eb386e0f89740337/rmd/ibd2015_nkcell.Rmd" target="_blank">b464c14</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-04
</td>
<td>
wflow_publish(“rmd/ibd2015_nkcell.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/SUwonglab/rss-net/2a4774000d0152f711f3fc39556ca48cc0757468/docs/ibd2015_nkcell.html" target="_blank">2a47740</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/SUwonglab/rss-net/blob/ea4499bee6140677fced879e9874830d8d533ff2/rmd/ibd2015_nkcell.Rmd" target="_blank">ea4499b</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-04
</td>
<td>
wflow_publish(“rmd/ibd2015_nkcell.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/SUwonglab/rss-net/e454742bb18d7b9a3045ab358fefe3547c01c5b1/docs/ibd2015_nkcell.html" target="_blank">e454742</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/SUwonglab/rss-net/blob/08f4c538cd3a32ac60fdf91273e45cb105b922de/rmd/ibd2015_nkcell.Rmd" target="_blank">08f4c53</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-04
</td>
<td>
wflow_publish(“rmd/ibd2015_nkcell.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/SUwonglab/rss-net/72d8e787989672e5603089720f6f961675824e1d/docs/ibd2015_nkcell.html" target="_blank">72d8e78</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/SUwonglab/rss-net/blob/6b97ae57677f1ecf48f6e4b9a33ee373a9b3c074/rmd/ibd2015_nkcell.Rmd" target="_blank">6b97ae5</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-04
</td>
<td>
wflow_publish(“rmd/ibd2015_nkcell.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/SUwonglab/rss-net/0e9e3a212cc95f2261c627c626031a6cea574580/docs/ibd2015_nkcell.html" target="_blank">0e9e3a2</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/SUwonglab/rss-net/blob/eb9c07307a05543199d274a3405f2b45fb1e95a0/rmd/ibd2015_nkcell.Rmd" target="_blank">eb9c073</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-03
</td>
<td>
wflow_publish(“rmd/ibd2015_nkcell.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/SUwonglab/rss-net/blob/6002e979a610d4d8ea57607b71763880abcadbb0/rmd/ibd2015_nkcell.Rmd" target="_blank">6002e97</a>
</td>
<td>
Xiang Zhu
</td>
<td>
2020-03-03
</td>
<td>
update ibd2015 nkcell example
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/SUwonglab/rss-net/6002e979a610d4d8ea57607b71763880abcadbb0/docs/ibd2015_nkcell.html" target="_blank">6002e97</a>
</td>
<td>
Xiang Zhu
</td>
<td>
2020-03-03
</td>
<td>
update ibd2015 nkcell example
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/SUwonglab/rss-net/5974d280da0308a0da739741c17bc3ecb45562fe/docs/ibd2015_nkcell.html" target="_blank">5974d28</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/SUwonglab/rss-net/blob/4a13f45807786aacfa63dc4ed0ca2e2067abcf76/rmd/ibd2015_nkcell.Rmd" target="_blank">4a13f45</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-03
</td>
<td>
wflow_publish(“rmd/ibd2015_nkcell.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/SUwonglab/rss-net/bf742e05134d16bf1f49e590bac893103bc7d473/docs/ibd2015_nkcell.html" target="_blank">bf742e0</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/SUwonglab/rss-net/blob/a16d453c7ab14ff07600ea8da1ec63c538d151b2/rmd/ibd2015_nkcell.Rmd" target="_blank">a16d453</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-02
</td>
<td>
wflow_publish(“rmd/ibd2015_nkcell.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/SUwonglab/rss-net/785a4239838914d0a47318f9190c42c4b6275ae2/docs/ibd2015_nkcell.html" target="_blank">785a423</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/SUwonglab/rss-net/blob/bda738237a2d508843d9cf3bf62634dc16b8c024/rmd/ibd2015_nkcell.Rmd" target="_blank">bda7382</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-01
</td>
<td>
wflow_publish(“rmd/ibd2015_nkcell.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/SUwonglab/rss-net/830b5816461415255fa4d8205853b11f6b0428e8/docs/ibd2015_nkcell.html" target="_blank">830b581</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/SUwonglab/rss-net/blob/a5b592e1a0fa19749acf3068b8666cfddbb735a9/rmd/ibd2015_nkcell.Rmd" target="_blank">a5b592e</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-01
</td>
<td>
wflow_publish(“rmd/ibd2015_nkcell.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/SUwonglab/rss-net/c3ca100647a4b76c9e74520dd555d3928f37e945/docs/ibd2015_nkcell.html" target="_blank">c3ca100</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-02-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/SUwonglab/rss-net/blob/2e5247ba3de7f13d0a9c0d2e4d2ee103cfd1fe07/rmd/ibd2015_nkcell.Rmd" target="_blank">2e5247b</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-02-29
</td>
<td>
wflow_publish(“rmd/ibd2015_nkcell.Rmd”)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="overview" class="section level2">
<h2>Overview</h2>
<p>Here we describe an end-to-end RSS-NET analysis of inflammatory bowel disease (IBD) GWAS summary statistics <a href="https://doi.org/10.1038/ng.3359">(Liu et al, 2015)</a> and a gene regulatory network inferred for natural killer (NK) cells <a href="https://doi.org/10.1038/nature11247">(ENCODE Project Consortium, 2012)</a>. This example illustrates the actual data analyses performed in <a href="xxx">Zhu et al (2020)</a>.</p>
<p>To reproduce results of this example, please use scripts in the directory <a href="https://github.com/SUwonglab/rss-net/tree/master/examples/ibd2015_nkcell"><code>ibd2015_nkcell/</code></a>, and follow the step-by-step guide below. Before running any script in <a href="https://github.com/SUwonglab/rss-net/tree/master/examples/ibd2015_nkcell"><code>ibd2015_nkcell/</code></a>, please <a href="setup.html">install</a> RSS-NET.</p>
<p>Since a real genome-wide analysis is conducted here, this example is more complicated than the previous <a href="wtccc_bcell.html">simulation example</a>. It is advisable to go through the previous <a href="wtccc_bcell.html">simulation example</a> before diving into the present real-world example.</p>
<p>With the software installed and the input data downloaded, one should be able to run this example by simply typing the following line in shell:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode zsh"><code class="sourceCode zsh"><a class="sourceLine" id="cb1-1" title="1">$ sbatch ibd2015_nkcell.sbatch</a></code></pre></div>
<p>If a different directory is used to store the software, input data and/or output data, please modify file paths in the given scripts accordingly.</p>
</div>
<div id="step-by-step-illustration" class="section level2">
<h2>Step-by-step illustration</h2>
<div id="download-data-files" class="section level3">
<h3>Download data files</h3>
<p>All data files required to run this example are freely available at Zenodo <a href="https://doi.org/10.5281/zenodo.3698240"><img src="https://zenodo.org/badge/DOI/10.5281/zenodo.3698240.svg" alt="DOI" /></a>. Please contact me (<code>xiangzhu[at]stanford.edu</code>) if you have any trouble accessing these files. After a complete download, you should see the following files.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode zsh"><code class="sourceCode zsh"><a class="sourceLine" id="cb2-1" title="1">$ tree <span class="kw">.</span> </a>
<a class="sourceLine" id="cb2-2" title="2"><span class="kw">.</span></a>
<a class="sourceLine" id="cb2-3" title="3">├── ibd2015_gene_grch37.mat</a>
<a class="sourceLine" id="cb2-4" title="4">├── ibd2015_nkcell_data.md5</a>
<a class="sourceLine" id="cb2-5" title="5">├── ibd2015_nkcell_full_results.zip</a>
<a class="sourceLine" id="cb2-6" title="6">├── ibd2015_nkcell_summary_results.zip</a>
<a class="sourceLine" id="cb2-7" title="7">├── ibd2015_NK_snp2gene_cis.mat</a>
<a class="sourceLine" id="cb2-8" title="8">├── ibd2015_null_seed_459_squarem_step2.mat</a>
<a class="sourceLine" id="cb2-9" title="9">├── ibd2015_Primary_Natural_Killer_cells_from_peripheral_blood_snp2net.mat</a>
<a class="sourceLine" id="cb2-10" title="10">├── ibd2015_snp2gene.mat</a>
<a class="sourceLine" id="cb2-11" title="11">├── ibd2015_sumstat.mat</a>
<a class="sourceLine" id="cb2-12" title="12">└── Primary_Natural_Killer_cells_from_peripheral_blood_gene2gene.mat</a>
<a class="sourceLine" id="cb2-13" title="13"></a>
<a class="sourceLine" id="cb2-14" title="14">0 directories, 10 files</a></code></pre></div>
<p>To help readers confirm if they use the same files as we do, we report 128-bit MD5 hashes of all files in <code>ibd2015_nkcell_data.md5</code>.</p>
<p>To help readers confirm if they can reproduce results of this example, we also provide the full results (<code>ibd2015_nkcell_full_results.zip</code>) and summarized results (<code>ibd2015_nkcell_summary_results.zip</code>) in the same Zenodo deposit.</p>
<p>For simplicity and generality, we introduce the following short-hand notations.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode matlab"><code class="sourceCode matlab"><a class="sourceLine" id="cb3-1" title="1">gwas = <span class="st">&#39;ibd2015&#39;</span>;</a>
<a class="sourceLine" id="cb3-2" title="2">net  = <span class="st">&#39;Primary_Natural_Killer_cells_from_peripheral_blood&#39;</span>; </a>
<a class="sourceLine" id="cb3-3" title="3">cis  = <span class="st">&#39;NK&#39;</span>;</a></code></pre></div>
<div id="gwas_sumstat.mat-processed-gwas-summary-statistics-and-ld-matrix-estimates" class="section level4">
<h4>1. <code>${gwas}_sumstat.mat</code>: processed GWAS summary statistics and LD matrix estimates</h4>
<p>This file contains processed IBD GWAS summary statistics and LD matrix estimates for 1.1 million common SNPs. This file is large (43G) because of the LD matrix .</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode zsh"><code class="sourceCode zsh"><a class="sourceLine" id="cb4-1" title="1">$ md5sum ibd2015_sumstat.mat</a>
<a class="sourceLine" id="cb4-2" title="2">ad1763079ee7e46b21722f74e037a230  ibd2015_sumstat.mat</a>
<a class="sourceLine" id="cb4-3" title="3"></a>
<a class="sourceLine" id="cb4-4" title="4">$ <span class="kw">du</span> -sh ibd2015_sumstat.mat                                             </a>
<a class="sourceLine" id="cb4-5" title="5">43G ibd2015_sumstat.mat</a></code></pre></div>
<p>Let’s look at the contents of <code>ibd2015_sumstat.mat</code>.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode matlab"><code class="sourceCode matlab"><a class="sourceLine" id="cb5-1" title="1">&gt;&gt; sumstat = matfile(<span class="st">&#39;ibd2015_sumstat.mat&#39;</span>);</a>
<a class="sourceLine" id="cb5-2" title="2">&gt;&gt; sumstat</a>
<a class="sourceLine" id="cb5-3" title="3">  matlab.io.MatFile</a>
<a class="sourceLine" id="cb5-4" title="4">  Properties:</a>
<a class="sourceLine" id="cb5-5" title="5">      Properties.Source: <span class="st">&#39;ibd2015_sumstat.mat&#39;</span></a>
<a class="sourceLine" id="cb5-6" title="6">    Properties.Writable: false                                                          </a>
<a class="sourceLine" id="cb5-7" title="7">                     BR: [<span class="fl">22</span>x1 cell]                                                    </a>
<a class="sourceLine" id="cb5-8" title="8">                      R: [<span class="fl">22</span>x1 cell]                                                    </a>
<a class="sourceLine" id="cb5-9" title="9">                  SiRiS: [<span class="fl">22</span>x1 cell]                                                    </a>
<a class="sourceLine" id="cb5-10" title="10">                betahat: [<span class="fl">22</span>x1 cell]                                                    </a>
<a class="sourceLine" id="cb5-11" title="11">                    chr: [<span class="fl">22</span>x1 cell]                                                    </a>
<a class="sourceLine" id="cb5-12" title="12">                    pos: [<span class="fl">22</span>x1 cell]                                                    </a>
<a class="sourceLine" id="cb5-13" title="13">                     se: [<span class="fl">22</span>x1 cell]</a></code></pre></div>
<p>GWAS summary statistics and LD estimates are stored as <a href="https://www.mathworks.com/help/matlab/cell-arrays.html">cell arrays</a>. RSS-NET only uses the following variables:</p>
<ul>
<li><code>betahat{j,1}</code>, single-SNP effect size estimates of all SNPs on chromosome <code>j</code>;</li>
<li><code>se{j,1}</code>, standard errors of <code>betahat{j, 1}</code>;</li>
<li><code>chr{j,1}</code> and <code>pos{j, 1}</code>, physical positions of these SNPs (GRCh37 build);</li>
<li><code>SiRiS{j,1}</code>, a <a href="https://www.mathworks.com/help/matlab/sparse-matrices.html">sparse matrix</a>, defined as <code>repmat((1./se),1,p) .* R .* repmat((1./se)',p,1)</code>, where <code>R</code> is the estimated LD matrix of these <code>p</code> SNPs.</li>
</ul>
</div>
<div id="gwas_snp2gene.mat-physical-distance-between-snps-and-genes" class="section level4">
<h4>2. <code>${gwas}_snp2gene.mat</code>: physical distance between SNPs and genes</h4>
<p>This file contains the physical distance between each GWAS SNP and each protein-coding gene, within 1 Mb. This file corresponds to <span class="math inline">\({\bf G}_j\)</span> in the RSS-NET model.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode zsh"><code class="sourceCode zsh"><a class="sourceLine" id="cb6-1" title="1">$ md5sum ibd2015_snp2gene.mat </a>
<a class="sourceLine" id="cb6-2" title="2">7832838e2675e4cf3b85f471fed95554  ibd2015_snp2gene.mat</a>
<a class="sourceLine" id="cb6-3" title="3"></a>
<a class="sourceLine" id="cb6-4" title="4">$ <span class="kw">du</span> -sh ibd2015_snp2gene.mat </a>
<a class="sourceLine" id="cb6-5" title="5">224M    ibd2015_snp2gene.mat</a></code></pre></div>
<p>In this example, there are 18334 genes and 1081481 SNPs.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode matlab"><code class="sourceCode matlab"><a class="sourceLine" id="cb7-1" title="1">&gt;&gt; snp2gene = matfile(<span class="st">&#39;ibd2015_snp2gene.mat&#39;</span>);</a>
<a class="sourceLine" id="cb7-2" title="2">&gt;&gt; snp2gene                                   </a>
<a class="sourceLine" id="cb7-3" title="3">  matlab.io.MatFile</a>
<a class="sourceLine" id="cb7-4" title="4">  Properties:</a>
<a class="sourceLine" id="cb7-5" title="5">      Properties.Source: <span class="st">&#39;ibd2015_snp2gene.mat&#39;</span></a>
<a class="sourceLine" id="cb7-6" title="6">    Properties.Writable: false                                                                </a>
<a class="sourceLine" id="cb7-7" title="7">                    chr: [<span class="fl">1081481</span>x1  int32]                                                   </a>
<a class="sourceLine" id="cb7-8" title="8">                  colid: [<span class="fl">14126805</span>x1 int32]                                                   </a>
<a class="sourceLine" id="cb7-9" title="9">                numgene: [<span class="fl">1</span>x1        int32]                                                   </a>
<a class="sourceLine" id="cb7-10" title="10">                 numsnp: [<span class="fl">1</span>x1        int32]                                                   </a>
<a class="sourceLine" id="cb7-11" title="11">                    pos: [<span class="fl">1081481</span>x1  int32]                                                   </a>
<a class="sourceLine" id="cb7-12" title="12">                  rowid: [<span class="fl">14126805</span>x1 int32]                                                   </a>
<a class="sourceLine" id="cb7-13" title="13">                    val: [<span class="fl">14126805</span>x1 double]</a>
<a class="sourceLine" id="cb7-14" title="14"></a>
<a class="sourceLine" id="cb7-15" title="15">&gt;&gt; [snp2gene.numgene snp2gene.numsnp]                                                              </a>
<a class="sourceLine" id="cb7-16" title="16">     <span class="fl">18334</span>   <span class="fl">1081481</span></a></code></pre></div>
<p>The SNP-to-gene distance information is captured by a three-column matrix <code>[colid rowid val]</code>. For example, the distance between gene <code>1</code> and SNP <code>6</code> is <code>978947</code> bps.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode matlab"><code class="sourceCode matlab"><a class="sourceLine" id="cb8-1" title="1">&gt;&gt; colid=snp2gene.colid; rowid=snp2gene.rowid; val=snp2gene.val;</a>
<a class="sourceLine" id="cb8-2" title="2">&gt;&gt; [colid(<span class="fl">6</span>) rowid(<span class="fl">6</span>) val(<span class="fl">6</span>)]</a>
<a class="sourceLine" id="cb8-3" title="3">        <span class="fl">1</span>        <span class="fl">6</span>   <span class="fl">978947</span></a></code></pre></div>
</div>
<div id="net_gene2gene.mat-gene-regulatory-network" class="section level4">
<h4>3. <code>${net}_gene2gene.mat</code>: gene regulatory network</h4>
<p>This file contains information of gene-to-gene connections in a given regulatory network.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode zsh"><code class="sourceCode zsh"><a class="sourceLine" id="cb9-1" title="1">$ md5sum Primary_Natural_Killer_cells_from_peripheral_blood_gene2gene.mat </a>
<a class="sourceLine" id="cb9-2" title="2">35ac724b86f7777d87116cc48166caa2  Primary_Natural_Killer_cells_from_peripheral_blood_gene2gene.mat</a>
<a class="sourceLine" id="cb9-3" title="3"></a>
<a class="sourceLine" id="cb9-4" title="4">$ <span class="kw">du</span> -sh Primary_Natural_Killer_cells_from_peripheral_blood_gene2gene.mat</a>
<a class="sourceLine" id="cb9-5" title="5">1.7M    Primary_Natural_Killer_cells_from_peripheral_blood_gene2gene.mat</a></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="sourceCode matlab"><code class="sourceCode matlab"><a class="sourceLine" id="cb10-1" title="1">&gt;&gt; gene2gene = matfile(<span class="st">&#39;Primary_Natural_Killer_cells_from_peripheral_blood_gene2gene.mat&#39;</span>);</a>
<a class="sourceLine" id="cb10-2" title="2">&gt;&gt; gene2gene</a>
<a class="sourceLine" id="cb10-3" title="3">  matlab.io.MatFile</a>
<a class="sourceLine" id="cb10-4" title="4">  Properties:</a>
<a class="sourceLine" id="cb10-5" title="5">      Properties.Source: <span class="st">&#39;Primary_Natural_Killer_cells_from_peripheral_blood_gene2gene.mat&#39;</span></a>
<a class="sourceLine" id="cb10-6" title="6">    Properties.Writable: false                                                                                                            </a>
<a class="sourceLine" id="cb10-7" title="7">                  colid: [<span class="fl">110733</span>x1 int32]                                                                                                 </a>
<a class="sourceLine" id="cb10-8" title="8">                numgene: [<span class="fl">1</span>x1      int32]                                                                                                 </a>
<a class="sourceLine" id="cb10-9" title="9">                  rowid: [<span class="fl">110733</span>x1 int32]                                                                                                 </a>
<a class="sourceLine" id="cb10-10" title="10">                    val: [<span class="fl">110733</span>x1 double]</a></code></pre></div>
<p>For implementation convenience, this file contains the trivial case where each gene is mapped to itself with <code>val=1</code>.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode matlab"><code class="sourceCode matlab"><a class="sourceLine" id="cb11-1" title="1">&gt;&gt; colid=gene2gene.colid; rowid=gene2gene.rowid; val=gene2gene.val;             </a>
<a class="sourceLine" id="cb11-2" title="2">&gt;&gt; [gene2gene.numgene sum(colid==rowid) unique(val(colid==rowid))]</a>
<a class="sourceLine" id="cb11-3" title="3">   <span class="fl">18334</span>   <span class="fl">18334</span>       <span class="fl">1</span></a></code></pre></div>
<p>For a given network, transcription factors (TFs) are stored in <code>rowid</code> and target genes (TGs) are stored in <code>colid</code>. In this example there are 3105 TGs and 376 TFs. Among these TFs and TGs, there are 92399 edges. The edge weights range from 0.61 to 1. These TF-to-TG connections and edge weights correspond to <span class="math inline">\(\{{\bf T}_g,v_{gt}\}\)</span> in the RSS-NET model.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode matlab"><code class="sourceCode matlab"><a class="sourceLine" id="cb12-1" title="1">&gt;&gt; [length(unique(colid(colid ~= rowid))) length(unique(rowid(colid ~= rowid)))]</a>
<a class="sourceLine" id="cb12-2" title="2">        <span class="fl">3105</span>         <span class="fl">376</span></a>
<a class="sourceLine" id="cb12-3" title="3">&gt;&gt; [length(colid(colid ~= rowid)) length(rowid(colid ~= rowid))]</a>
<a class="sourceLine" id="cb12-4" title="4">       <span class="fl">92399</span>       <span class="fl">92399</span></a>
<a class="sourceLine" id="cb12-5" title="5"></a>
<a class="sourceLine" id="cb12-6" title="6">&gt;&gt; val_tftg = val(colid ~= rowid);</a>
<a class="sourceLine" id="cb12-7" title="7">&gt;&gt; [min(val_tftg) quantile(val_tftg, <span class="fl">0.25</span>) median(val_tftg) quantile(val_tftg, <span class="fl">0.75</span>) max(val_tftg)]</a>
<a class="sourceLine" id="cb12-8" title="8">    <span class="fl">0.6138</span>    <span class="fl">0.6324</span>    <span class="fl">0.6568</span>    <span class="fl">0.6949</span>    <span class="fl">1.0000</span>       </a></code></pre></div>
</div>
<div id="gwas_net_snp2net.mat-snp-to-network-proximity-annotation" class="section level4">
<h4>4. <code>${gwas}_${net}_snp2net.mat</code>: SNP-to-network proximity annotation</h4>
<p>This file contains binary annotations whether a SNP is “near” the given network, that is, within 100 kb of any network element (TF, TG or associated regulatory elements).</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode zsh"><code class="sourceCode zsh"><a class="sourceLine" id="cb13-1" title="1">$ md5sum ibd2015_Primary_Natural_Killer_cells_from_peripheral_blood_snp2net.mat</a>
<a class="sourceLine" id="cb13-2" title="2">d96cd9b32759f954cc37680dc6aeafd8  ibd2015_Primary_Natural_Killer_cells_from_peripheral_blood_snp2net.mat</a>
<a class="sourceLine" id="cb13-3" title="3"></a>
<a class="sourceLine" id="cb13-4" title="4">$ <span class="kw">du</span> -sh ibd2015_Primary_Natural_Killer_cells_from_peripheral_blood_snp2net.mat</a>
<a class="sourceLine" id="cb13-5" title="5">21M ibd2015_Primary_Natural_Killer_cells_from_peripheral_blood_snp2net.mat</a></code></pre></div>
<p>In this example, there are 1081481 GWAS SNPs and 382443 of them are near the NK cell network (i.e. <code>val=1</code>).</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode matlab"><code class="sourceCode matlab"><a class="sourceLine" id="cb14-1" title="1">&gt;&gt; snp2net = matfile(<span class="st">&#39;ibd2015_Primary_Natural_Killer_cells_from_peripheral_blood_snp2net.mat&#39;</span>);</a>
<a class="sourceLine" id="cb14-2" title="2">&gt;&gt; snp2net</a>
<a class="sourceLine" id="cb14-3" title="3">  matlab.io.MatFile</a>
<a class="sourceLine" id="cb14-4" title="4">  Properties:</a>
<a class="sourceLine" id="cb14-5" title="5">      Properties.Source: <span class="st">&#39;ibd2015_Primary_Natural_Killer_cells_from_peripheral_blood_snp2net.mat&#39;</span></a>
<a class="sourceLine" id="cb14-6" title="6">    Properties.Writable: false</a>
<a class="sourceLine" id="cb14-7" title="7">                    chr: [<span class="fl">1081481</span>x1 int32]</a>
<a class="sourceLine" id="cb14-8" title="8">                    pos: [<span class="fl">1081481</span>x1 int32]</a>
<a class="sourceLine" id="cb14-9" title="9">                  snpid: [<span class="fl">1081481</span>x1 int32]</a>
<a class="sourceLine" id="cb14-10" title="10">                    val: [<span class="fl">1081481</span>x1 double]</a>
<a class="sourceLine" id="cb14-11" title="11">                 window: [<span class="fl">1</span>x1       double]</a>
<a class="sourceLine" id="cb14-12" title="12">                 </a>
<a class="sourceLine" id="cb14-13" title="13">&gt;&gt; [length(snp2net.val) sum(snp2net.val) snp2net.window]</a>
<a class="sourceLine" id="cb14-14" title="14">     <span class="fl">1081481</span>      <span class="fl">382443</span>      <span class="fl">100000</span></a>
<a class="sourceLine" id="cb14-15" title="15">&gt;&gt; unique(snp2net.val)&#39;</a>
<a class="sourceLine" id="cb14-16" title="16">     <span class="fl">0</span>     <span class="fl">1</span>     </a></code></pre></div>
</div>
<div id="gwas_cis_snp2gene_cis.mat-snp-to-gene-cis-regulation" class="section level4">
<h4>5. <code>${gwas}_${cis}_snp2gene_cis.mat</code>: SNP-to-gene cis regulation</h4>
<p>This file contains the SNP-to-gene cis regulation scores derived from context-matching cis eQTL studies. This file corresponds to <span class="math inline">\((c_{jg}-1)\)</span> in the RSS-NET model.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode zsh"><code class="sourceCode zsh"><a class="sourceLine" id="cb15-1" title="1">$ md5sum ibd2015_NK_snp2gene_cis.mat</a>
<a class="sourceLine" id="cb15-2" title="2">dedad8e25773fad69576dbce0f7d9f93  ibd2015_NK_snp2gene_cis.mat</a>
<a class="sourceLine" id="cb15-3" title="3"></a>
<a class="sourceLine" id="cb15-4" title="4">$ <span class="kw">du</span> -sh ibd2015_NK_snp2gene_cis.mat</a>
<a class="sourceLine" id="cb15-5" title="5">165M    ibd2015_NK_snp2gene_cis.mat</a></code></pre></div>
<div class="sourceCode" id="cb16"><pre class="sourceCode matlab"><code class="sourceCode matlab"><a class="sourceLine" id="cb16-1" title="1">&gt;&gt; snp2gene_cis = matfile(<span class="st">&#39;ibd2015_NK_snp2gene_cis.mat&#39;</span>);</a>
<a class="sourceLine" id="cb16-2" title="2">&gt;&gt; snp2gene_cis</a>
<a class="sourceLine" id="cb16-3" title="3">  matlab.io.MatFile</a>
<a class="sourceLine" id="cb16-4" title="4">  Properties:</a>
<a class="sourceLine" id="cb16-5" title="5">      Properties.Source: <span class="st">&#39;ibd2015_NK_snp2gene_cis.mat&#39;</span></a>
<a class="sourceLine" id="cb16-6" title="6">    Properties.Writable: false</a>
<a class="sourceLine" id="cb16-7" title="7">                  colid: [<span class="fl">10790012</span>x1 int32]</a>
<a class="sourceLine" id="cb16-8" title="8">                numgene: [<span class="fl">1</span>x1        int32]</a>
<a class="sourceLine" id="cb16-9" title="9">                 numsnp: [<span class="fl">1</span>x1        int32]</a>
<a class="sourceLine" id="cb16-10" title="10">                  rowid: [<span class="fl">10790012</span>x1 int32]</a>
<a class="sourceLine" id="cb16-11" title="11">                    val: [<span class="fl">10790012</span>x1 double]</a></code></pre></div>
<p>In this example, there are 10790012 SNP-gene pairs with cis regulation scores available, consisting of 829280 SNPs and 18230 genes. The cis regulation scores (<code>val</code>) range from 0 to 0.76. The cis regulation scores used in this example are derived from recently published cis eQTLs in NK cells (<a href="https://doi.org/10.1016/j.cell.2018.10.022">Schmiedel et al, 2018</a>).</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode matlab"><code class="sourceCode matlab"><a class="sourceLine" id="cb17-1" title="1">&gt;&gt; [snp2gene_cis.numsnp length(unique(snp2gene_cis.rowid)) snp2gene_cis.numgene length(unique(snp2gene_cis.colid))]</a>
<a class="sourceLine" id="cb17-2" title="2">   <span class="fl">1081481</span>    <span class="fl">829280</span>     <span class="fl">18334</span>     <span class="fl">18230</span></a>
<a class="sourceLine" id="cb17-3" title="3">   </a>
<a class="sourceLine" id="cb17-4" title="4">&gt;&gt; val=snp2gene_cis.val;</a>
<a class="sourceLine" id="cb17-5" title="5">&gt;&gt; [min(val) quantile(val, <span class="fl">0.25</span>) median(val) quantile(val, <span class="fl">0.75</span>) max(val)]</a>
<a class="sourceLine" id="cb17-6" title="6">         <span class="fl">0</span>    <span class="fl">0.0226</span>    <span class="fl">0.0636</span>    <span class="fl">0.1172</span>    <span class="fl">0.7622</span>   </a></code></pre></div>
</div>
</div>
<div id="run-rss-net-analysis" class="section level3">
<h3>Run RSS-NET analysis</h3>
<p>To facilitate running RSS-NET on real data, we provide a generic script <a href="https://github.com/SUwonglab/rss-net/blob/master/examples/ibd2015_nkcell/analysis_template.m"><code>analysis_template.m</code></a>. For the present example, the RSS-NET analysis is implemented by <a href="https://github.com/SUwonglab/rss-net/blob/master/examples/ibd2015_nkcell/ibd2015_nkcell.m"><code>ibd2015_nkcell.m</code></a> and <a href="https://github.com/SUwonglab/rss-net/blob/master/examples/ibd2015_nkcell/ibd2015_nkcell.sbatch"><code>ibd2015_nkcell.sbatch</code></a>.</p>
<div id="specify-analysis-specific-variables" class="section level4">
<h4>1. Specify analysis-specific variables</h4>
<p>We need to specify a few analysis-specific variables that are required by <a href="https://github.com/SUwonglab/rss-net/blob/master/examples/ibd2015_nkcell/analysis_template.m"><code>analysis_template.m</code></a>, a template script that fits RSS-NET to the given data. For this example, we use <a href="https://github.com/SUwonglab/rss-net/blob/master/examples/ibd2015_nkcell/ibd2015_nkcell.m"><code>ibd2015_nkcell.m</code></a> for the specification. In brief, the following variables are specified.</p>
<ul>
<li>Data names: <code>gwas_name</code>, <code>net_name</code>, <code>cis_name</code>;</li>
<li>GWAS sample size and number of genes: <code>nsam</code>, <code>ngene</code>;</li>
<li>Hyper-parameter grid: <code>eta_set</code>, <code>rho_set</code>, <code>theta0_set</code>, <code>theta_set</code>.</li>
</ul>
<p>In general, if you want to use RSS-NET to analyze a different GWAS and/or network, simply modify <a href="https://github.com/SUwonglab/rss-net/blob/master/examples/ibd2015_nkcell/ibd2015_nkcell.m"><code>ibd2015_nkcell.m</code></a> and there is no need to change <a href="https://github.com/SUwonglab/rss-net/blob/master/examples/ibd2015_nkcell/analysis_template.m"><code>analysis_template.m</code></a>.</p>
</div>
<div id="submit-job-arrays" class="section level4">
<h4>2. Submit job arrays</h4>
<p>For a given GWAS and a given regulatory network, all RSS-NET analysis tasks are almost identical and they only differs in hyper-parameter values. To exploit this feature, we run one RSS-NET analysis as a <a href="https://slurm.schedmd.com/job_array.html">job array</a> with multiple tasks that run in parallel.</p>
<p>To this end, we write a simple sbatch script <a href="https://github.com/SUwonglab/rss-net/blob/master/examples/ibd2015_nkcell/ibd2015_nkcell.sbatch"><code>ibd2015_nkcell.sbatch</code></a>, and submit it to a cluster with <a href="https://slurm.schedmd.com/"><code>Slurm</code></a> available.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode zsh"><code class="sourceCode zsh"><a class="sourceLine" id="cb18-1" title="1">$ sbatch ibd2015_nkcell.sbatch</a></code></pre></div>
<p>After the submission, 125 tasks are created. As shown below, multiple tasks should run in different nodes simultaneously.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode zsh"><code class="sourceCode zsh"><a class="sourceLine" id="cb19-1" title="1">$ squeue -u xiangzhu</a>
<a class="sourceLine" id="cb19-2" title="2">             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST<span class="kw">(</span>REASON<span class="kw">)</span></a>
<a class="sourceLine" id="cb19-3" title="3">      62554249_107    owners ibd2015_ xiangzhu  R       0:32      1 sh02-17n12</a>
<a class="sourceLine" id="cb19-4" title="4">      62554249_108    owners ibd2015_ xiangzhu  R       0:32      1 sh02-17n12</a>
<a class="sourceLine" id="cb19-5" title="5">      62554249_109    owners ibd2015_ xiangzhu  R       0:32      1 sh01-28n08</a>
<a class="sourceLine" id="cb19-6" title="6">      62554249_110    owners ibd2015_ xiangzhu  R       0:32      1 sh01-17n18</a>
<a class="sourceLine" id="cb19-7" title="7">      62554249_111    owners ibd2015_ xiangzhu  R       0:32      1 sh01-26n33</a>
<a class="sourceLine" id="cb19-8" title="8">      62554249_112    owners ibd2015_ xiangzhu  R       0:32      1 sh01-27n30</a></code></pre></div>
<p>For each task of this job array, we request 1 node with 8 CPUs and 32 Gb total memory and set the maximum job wall-clock time as 12.5 hours (see <a href="https://github.com/SUwonglab/rss-net/blob/master/examples/ibd2015_nkcell/ibd2015_nkcell.sbatch"><code>ibd2015_nkcell.sbatch</code></a> for details). The actual memory utilized per task is 26.58 GB (efficiency: 85.06% of 31.25 GB). Across all 125 tasks, the actual running time per task ranges from 5 minutes to 8.9 hours, with median being 2.1 hours.</p>
<p>We request 8 CPUs for each task because RSS-NET takes advantage of <a href="https://www.mathworks.com/help/parallel-computing/parfor.html"><code>parfor</code></a> in the <a href="https://www.mathworks.com/help/distcomp/index.html">MATLAB Parallel Computing Toolbox</a>. If this toolbox is not available in your environment, you can still run the same RSS-NET codes on this example (in a serial manner), with longer computation time per task.</p>
<p>Each task of the job array saves results in a <a href="https://www.mathworks.com/help/matlab/import_export/mat-file-versions.html">Version 7 MAT-file</a>, <code>${gwas}_${net}_${cis}_out_${id}.mat</code>. Each MAT-file contains variational estimates for a given set of hyper-parameter values. For example, the following MAT-file <code>ibd2015_Primary_Natural_Killer_cells_from_peripheral_blood_NK_out_66.mat</code> stores RSS-NET results based on the 66-th set of hyper-parameter values from the grid.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode matlab"><code class="sourceCode matlab"><a class="sourceLine" id="cb20-1" title="1">&gt;&gt; res = matfile(<span class="st">&#39;ibd2015_Primary_Natural_Killer_cells_from_peripheral_blood_NK_out_66.mat&#39;</span>);</a>
<a class="sourceLine" id="cb20-2" title="2">&gt;&gt; res</a>
<a class="sourceLine" id="cb20-3" title="3">  matlab.io.MatFile</a>
<a class="sourceLine" id="cb20-4" title="4">  Properties:</a>
<a class="sourceLine" id="cb20-5" title="5">      Properties.Source: <span class="st">&#39;ibd2015_Primary_Natural_Killer_cells_from_peripheral_blood_NK_out_66.mat&#39;</span></a>
<a class="sourceLine" id="cb20-6" title="6">    Properties.Writable: false                                                                                                                            </a>
<a class="sourceLine" id="cb20-7" title="7">                  alpha: [<span class="fl">1081481</span>x1 double]                                                                                                               </a>
<a class="sourceLine" id="cb20-8" title="8">                   logw: [<span class="fl">1</span>x1       double]                                                                                                               </a>
<a class="sourceLine" id="cb20-9" title="9">                     mu: [<span class="fl">1081481</span>x1 double]                                                                                                               </a>
<a class="sourceLine" id="cb20-10" title="10">               run_time: [<span class="fl">1</span>x1       double]                                                                                                               </a>
<a class="sourceLine" id="cb20-11" title="11">                      s: [<span class="fl">1081481</span>x1 double]                                                                                                               </a>
<a class="sourceLine" id="cb20-12" title="12">                   sigb: [<span class="fl">1</span>x1       double]                                                                                                               </a>
<a class="sourceLine" id="cb20-13" title="13">                   sige: [<span class="fl">1</span>x1       double]                                                                                                               </a>
<a class="sourceLine" id="cb20-14" title="14">                  theta: [<span class="fl">1</span>x1       double]                                                                                                               </a>
<a class="sourceLine" id="cb20-15" title="15">                 theta0: [<span class="fl">1</span>x1       double]</a></code></pre></div>
<p>Here <code>[alpha mu s]</code> correspond to the optimal variational parameters <span class="math inline">\(\{\alpha_j^\star,\mu_j^\star,(\tau_j^\star)^2\}\)</span> for the given hyper-parameters, <code>logw</code> corresponds to the variational lower bound <span class="math inline">\(F^\star\)</span> and <code>[theta0 theta sigb sige]</code> corresponds to <span class="math inline">\(\{\theta_0,\theta,\sigma_0,\sigma\}\)</span>. Please see <a href="xxx">Supplementary Notes</a> of <a href="xxx">Zhu et al (2020)</a> for definitions.</p>
</div>
</div>
<div id="summarize-rss-net-results" class="section level3">
<h3>Summarize RSS-NET results</h3>
<p>The RSS-NET result files <code>${gwas}_${net}_${cis}_out_${id}.mat</code> shown above can be further summarized into both network-level and gene-level statistics as reported in <a href="xxx">Zhu et al (2020)</a>. To facilitate summarizing RSS-NET results, we provide a generic script <a href="https://github.com/SUwonglab/rss-net/blob/master/examples/ibd2015_nkcell/summary_template.m"><code>summary_template.m</code></a>. For the present example, the RSS-NET summary is implemented by <a href="https://github.com/SUwonglab/rss-net/blob/master/examples/ibd2015_nkcell/summarize_ibd2015_nkcell.m"><code>summarize_ibd2015_nkcell.m</code></a>. If you need to summarize a different RSS-NET analysis, simply modify <a href="https://github.com/SUwonglab/rss-net/blob/master/examples/ibd2015_nkcell/summarize_ibd2015_nkcell.m"><code>summarize_ibd2015_nkcell.m</code></a> and there is no need to change <a href="https://github.com/SUwonglab/rss-net/blob/master/examples/ibd2015_nkcell/summary_template.m"><code>summary_template.m</code></a>.</p>
<p>For this example, simply run the following line in a Matlab console.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode matlab"><code class="sourceCode matlab"><a class="sourceLine" id="cb21-1" title="1">&gt;&gt; run summarize_ibd2015_nkcell.m;</a></code></pre></div>
<p>Running <a href="https://github.com/SUwonglab/rss-net/blob/master/examples/ibd2015_nkcell/summarize_ibd2015_nkcell.m"><code>summarize_ibd2015_nkcell.m</code></a> yields two (much smaller) MAT-files: <code>${gwas}_${net}_${cis}_results_model.mat</code> that stores network-level enrichment results, and <code>${gwas}_${net}_${cis}_results_gene.mat</code> that stores locus-level association results.</p>
<div id="gwas_net_cis_results_model.mat-network-level-enrichments" class="section level4">
<h4>1. <code>${gwas}_${net}_${cis}_results_model.mat</code>: network-level enrichments</h4>
<p>To assess whether a regulatory network is enriched for genetic associations with a trait, we evaluate a Bayes factor (BF) comparing the baseline model (<span class="math inline">\(M_0:\theta=0~\text{and}~\sigma^2=0\)</span>) in RSS-NET with an enrichment model.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode matlab"><code class="sourceCode matlab"><a class="sourceLine" id="cb22-1" title="1">&gt;&gt; model_res = matfile(<span class="st">&#39;ibd2015_Primary_Natural_Killer_cells_from_peripheral_blood_NK_results_modet&#39;</span>);</a>
<a class="sourceLine" id="cb22-2" title="2">&gt;&gt; model_res</a>
<a class="sourceLine" id="cb22-3" title="3">  matlab.io.MatFile</a>
<a class="sourceLine" id="cb22-4" title="4">  Properties:</a>
<a class="sourceLine" id="cb22-5" title="5">      Properties.Source: <span class="st">&#39;ibd2015_Primary_Natural_Killer_cells_from_peripheral_blood_NK_results_model.mat&#39;</span></a>
<a class="sourceLine" id="cb22-6" title="6">    Properties.Writable: false                                                                                                                                                    </a>
<a class="sourceLine" id="cb22-7" title="7">               log10_bf: [<span class="fl">1</span>x1   double]                                                                                                                                           </a>
<a class="sourceLine" id="cb22-8" title="8">            log10_bf_ns: [<span class="fl">1</span>x1   double]                                                                                                                                           </a>
<a class="sourceLine" id="cb22-9" title="9">            log10_bf_nt: [<span class="fl">1</span>x1   double]                                                                                                                                           </a>
<a class="sourceLine" id="cb22-10" title="10">            log10_bf_ts: [<span class="fl">1</span>x1   double]                                                                                                                                           </a>
<a class="sourceLine" id="cb22-11" title="11">                   logw: [<span class="fl">125</span>x1 double]                                                                                                                                           </a>
<a class="sourceLine" id="cb22-12" title="12">                   sigb: [<span class="fl">125</span>x1 double]                                                                                                                                           </a>
<a class="sourceLine" id="cb22-13" title="13">                   sige: [<span class="fl">125</span>x1 double]                                                                                                                                           </a>
<a class="sourceLine" id="cb22-14" title="14">                  theta: [<span class="fl">125</span>x1 double]                                                                                                                                           </a>
<a class="sourceLine" id="cb22-15" title="15">                 theta0: [<span class="fl">125</span>x1 double]                                                                                                                                           </a>
<a class="sourceLine" id="cb22-16" title="16">                   time: [<span class="fl">125</span>x1 double]</a></code></pre></div>
<p>Here <code>log10_bf*</code> are log 10 BFs comparing the following 4 enrichment models against <span class="math inline">\(M_0\)</span>.</p>
<ul>
<li><code>log10_bf</code>: <span class="math inline">\(M_1:\theta&gt;0~\text{or}~\sigma^2&gt;0\)</span>;</li>
<li><code>log10_bf_ns</code>: <span class="math inline">\(M_{11}:\theta&gt;0~\text{and}~\sigma^2=0\)</span>;</li>
<li><code>log10_bf_nt</code>: <span class="math inline">\(M_{12}:\theta=0~\text{and}~\sigma^2&gt;0\)</span>;</li>
<li><code>log10_bf_ts</code>: <span class="math inline">\(M_{13}:\theta&gt;0~\text{and}~\sigma^2&gt;0\)</span>.</li>
</ul>
<p>Because <span class="math inline">\(M_1\)</span> is more flexible than other models, we mainly use <code>log10_bf</code> as recommended by <a href="xxx">Zhu et al (2020)</a>.</p>
<p>By running this example, we reproduce the enrichment BFs of <a href="https://xiangzhu.github.io/rss-peca/ibd2015_net.html">IBD GWAS and NK cell network</a> reported in <a href="xxx">Zhu et al (2020)</a>. The NK cell network shows strong enrichment of IBD genetic associations, which seems consistent with the role of NK cell in autoimmune diseases like IBD.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode matlab"><code class="sourceCode matlab"><a class="sourceLine" id="cb23-1" title="1">&gt;&gt; load ibd2015_Primary_Natural_Killer_cells_from_peripheral_blood_NK_results_model.mat</a>
<a class="sourceLine" id="cb23-2" title="2">&gt;&gt; [log10_bf log10_bf_ns log10_bf_nt log10_bf_ts]</a>
<a class="sourceLine" id="cb23-3" title="3">   <span class="fl">35.7048</span>   <span class="fl">29.4216</span>   <span class="fl">15.1461</span>   <span class="fl">35.8986</span></a></code></pre></div>
<p>Let’s perform a more rigorous check of reproducibility. For the same hyper-parameter values, we compare the resulting variational lower bounds from my previous run and from the current run. Differences are numerical negligible.</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode matlab"><code class="sourceCode matlab"><a class="sourceLine" id="cb24-1" title="1">&gt;&gt; res_file = <span class="st">&#39;ibd2015_Primary_Natural_Killer_cells_from_peripheral_blood_NK_results_model.mat&#39;</span>;               </a>
<a class="sourceLine" id="cb24-2" title="2">&gt;&gt; old_path = <span class="st">&#39;~/Dropbox/rss/Data/peca_human/job_camp/gwas33_net71/results/model_results/peca_encode/ibd2015/&#39;</span>;</a>
<a class="sourceLine" id="cb24-3" title="3">&gt;&gt; new_path = strcat(src_path,<span class="st">&#39;rss-net/examples/ibd2015_nkcell/results/&#39;</span>);                                             </a>
<a class="sourceLine" id="cb24-4" title="4">&gt;&gt; old_res = matfile(strcat(old_path,res_file));                                                               </a>
<a class="sourceLine" id="cb24-5" title="5">&gt;&gt; new_res = matfile(strcat(new_path,res_file));</a>
<a class="sourceLine" id="cb24-6" title="6"></a>
<a class="sourceLine" id="cb24-7" title="7">&gt;&gt; [min(old_res.logw - new_res.logw) median(old_res.logw - new_res.logw) max(old_res.logw - new_res.logw)]</a>
<a class="sourceLine" id="cb24-8" title="8">   <span class="fl">1.0e-10</span> *</a>
<a class="sourceLine" id="cb24-9" title="9">   -<span class="fl">0.0568</span>         <span class="fl">0</span>    <span class="fl">0.1592</span></a></code></pre></div>
</div>
<div id="gwas_net_cis_results_gene.mat-locus-level-associations" class="section level4">
<h4>2. <code>${gwas}_${net}_${cis}_results_gene.mat</code>: locus-level associations</h4>
<p>To summarize association between a locus and a trait, we compute <span class="math inline">\(P_1^{\sf net}\)</span>, the posterior probability that at least one SNP <span class="math inline">\(j\)</span> in the locus is associated with the trait (<span class="math inline">\(\beta_j\neq 0\)</span>): <span class="math display">\[
P_1^{\sf net}=1-\Pr(\beta_j=0,~\forall j\in\text{locus}~|~\text{data},M_1).
\]</span> As in <a href="xxx">Zhu et al (2020)</a>, a locus is defined as the transcribed region of a gene plus 100 kb upstream and downstream. The locus definition is provided in <code>ibd2015_gene_grch37.mat</code>.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode matlab"><code class="sourceCode matlab"><a class="sourceLine" id="cb25-1" title="1">&gt;&gt; gene_res=matfile(<span class="st">&#39;ibd2015_Primary_Natural_Killer_cells_from_peripheral_blood_NK_results_gene.mat&#39;</span>);</a>
<a class="sourceLine" id="cb25-2" title="2">&gt;&gt; gene_res</a>
<a class="sourceLine" id="cb25-3" title="3">  matlab.io.MatFile</a>
<a class="sourceLine" id="cb25-4" title="4">  Properties:</a>
<a class="sourceLine" id="cb25-5" title="5">      Properties.Source: <span class="st">&#39;ibd2015_Primary_Natural_Killer_cells_from_peripheral_blood_NK_results_gene.mat&#39;</span></a>
<a class="sourceLine" id="cb25-6" title="6">    Properties.Writable: false                                                                                                                                                   </a>
<a class="sourceLine" id="cb25-7" title="7">                P1_gene: [<span class="fl">18334</span>x1 double]                                                                                                                                        </a>
<a class="sourceLine" id="cb25-8" title="8">               gene_chr: [<span class="fl">18334</span>x1 double]                                                                                                                                        </a>
<a class="sourceLine" id="cb25-9" title="9">             gene_start: [<span class="fl">18334</span>x1 double]                                                                                                                                        </a>
<a class="sourceLine" id="cb25-10" title="10">              gene_stop: [<span class="fl">18334</span>x1 double]</a></code></pre></div>
<p>Here <code>P1_gene</code> corresponds to <span class="math inline">\(P_1^{\sf net}\)</span> and <code>[gene_chr gene_start gene_stop]</code> denote physical position of genes based on GRCh37.</p>
<p>Let’s perform a reproducibility check for gene-level results. Again, my previous analysis and the current analysis yield numerically identical answers.</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode matlab"><code class="sourceCode matlab"><a class="sourceLine" id="cb26-1" title="1">&gt;&gt; res_file = <span class="st">&#39;ibd2015_Primary_Natural_Killer_cells_from_peripheral_blood_NK_results_gene.mat&#39;</span>; </a>
<a class="sourceLine" id="cb26-2" title="2">&gt;&gt; old_path = <span class="st">&#39;~/Dropbox/rss/Data/peca_human/job_camp/gwas33_net71/results/gene_results/peca_encode/ibd2015/&#39;</span>; </a>
<a class="sourceLine" id="cb26-3" title="3">&gt;&gt; new_path = strcat(src_path,<span class="st">&#39;rss-net/examples/ibd2015_nkcell/results/&#39;</span>);</a>
<a class="sourceLine" id="cb26-4" title="4">&gt;&gt; old_res = matfile(strcat(old_path,res_file));                                                              </a>
<a class="sourceLine" id="cb26-5" title="5">&gt;&gt; new_res = matfile(strcat(new_path,res_file));</a>
<a class="sourceLine" id="cb26-6" title="6"></a>
<a class="sourceLine" id="cb26-7" title="7">&gt;&gt; [min(old_res.P1_gene - new_res.P1_gene) median(old_res.P1_gene - new_res.P1_gene) max(old_res.P1_gene - new_res.P1_gene)]                  </a>
<a class="sourceLine" id="cb26-8" title="8">   <span class="fl">1.0e-15</span> *</a>
<a class="sourceLine" id="cb26-9" title="9">   -<span class="fl">0.2220</span>    <span class="fl">0.0278</span>    <span class="fl">0.5551</span></a></code></pre></div>
</div>
</div>
</div>
<div id="more-examples" class="section level2">
<h2>More examples</h2>
<p>The RSS-NET analyses of 18 complex traits and 38 gene regulatory networks reported in <a href="xxx">Zhu et al (2020)</a> are essentially 684 modified reruns of the example above (with different input GWAS and/or network data, and different hyper-parameter grids). Our full analysis results are publicly available at <a href="https://xiangzhu.github.io/rss-peca" class="uri">https://xiangzhu.github.io/rss-peca</a>.</p>
</div>
<div id="appendix" class="section level2">
<h2>Appendix</h2>
<p>Careful readers may notice an optional input data file <code>ibd2015_null_seed_459_squarem_step2.mat</code> specified in <a href="https://github.com/SUwonglab/rss-net/blob/master/examples/ibd2015_nkcell/ibd2015_nkcell.m"><code>ibd2015_nkcell.m</code></a> and used in <a href="https://github.com/SUwonglab/rss-net/blob/master/examples/ibd2015_nkcell/analysis_template.m"><code>analysis_template.m</code></a>. This file provides the <a href="https://doi.org/10.1038/s41467-018-06805-x">RSS-E baseline model</a> fitting results of IBD GWAS data. If this file is available, <a href="https://github.com/SUwonglab/rss-net/blob/master/examples/ibd2015_nkcell/analysis_template.m"><code>analysis_template.m</code></a> uses the optimal <a href="https://doi.org/10.1038/s41467-018-06805-x">RSS-E baseline model</a> results to initialize RSS-NET. If not, RSS-NET uses a random initialization.</p>
<p>For this example, here are the enrichment BFs based on using the optimal <a href="https://doi.org/10.1038/s41467-018-06805-x">RSS-E baseline model</a> results to initialize RSS-NET.</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode matlab"><code class="sourceCode matlab"><a class="sourceLine" id="cb27-1" title="1">&gt;&gt; [log10_bf log10_bf_ns log10_bf_nt log10_bf_ts]</a>
<a class="sourceLine" id="cb27-2" title="2">   <span class="fl">35.7048</span>   <span class="fl">29.4216</span>   <span class="fl">15.1461</span>   <span class="fl">35.8986</span></a></code></pre></div>
<p>Here are the enrichment BFs based on random initialization, which are consistent with, but smaller than previous results.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode matlab"><code class="sourceCode matlab"><a class="sourceLine" id="cb28-1" title="1">&gt;&gt; [log10_bf log10_bf_ns log10_bf_nt log10_bf_ts]</a>
<a class="sourceLine" id="cb28-2" title="2">   <span class="fl">32.1495</span>   <span class="fl">29.5295</span>   <span class="fl">11.0758</span>   <span class="fl">32.3432</span></a></code></pre></div>
<p>Using optimal <a href="https://doi.org/10.1038/s41467-018-06805-x">RSS-E baseline model</a> results to initialize RSS-NET is not required, but highly recommended in practice, because this often yields a better fit as shown above. Please see <a href="https://stephenslab.github.io/rss/Example-5B">this tutorial</a> for more details of fitting <a href="https://doi.org/10.1038/s41467-018-06805-x">RSS-E baseline model</a> on GWAS data.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb29-1" title="1">devtools<span class="op">::</span><span class="kw">session_info</span>()</a></code></pre></div>
<pre><code>─ Session info ───────────────────────────────────────────────────────────────
 setting  value                       
 version  R version 3.6.3 (2020-02-29)
 os       Ubuntu 18.04.4 LTS          
 system   x86_64, linux-gnu           
 ui       X11                         
 language (EN)                        
 collate  en_US.UTF-8                 
 ctype    en_US.UTF-8                 
 tz       America/Los_Angeles         
 date     2020-03-09                  

─ Packages ───────────────────────────────────────────────────────────────────
 package     * version date       lib source        
 assertthat    0.2.1   2019-03-21 [1] CRAN (R 3.6.0)
 backports     1.1.5   2019-10-02 [1] CRAN (R 3.6.1)
 callr         3.4.2   2020-02-12 [1] CRAN (R 3.6.3)
 cli           2.0.2   2020-02-28 [1] CRAN (R 3.6.3)
 crayon        1.3.4   2017-09-16 [1] CRAN (R 3.6.0)
 desc          1.2.0   2018-05-01 [1] CRAN (R 3.6.0)
 devtools      2.2.2   2020-02-17 [1] CRAN (R 3.6.3)
 digest        0.6.25  2020-02-23 [1] CRAN (R 3.6.3)
 ellipsis      0.3.0   2019-09-20 [1] CRAN (R 3.6.1)
 evaluate      0.14    2019-05-28 [1] CRAN (R 3.6.0)
 fansi         0.4.1   2020-01-08 [1] CRAN (R 3.6.2)
 fs            1.3.2   2020-03-05 [1] CRAN (R 3.6.3)
 git2r         0.26.1  2019-06-29 [1] CRAN (R 3.6.0)
 glue          1.3.1   2019-03-12 [1] CRAN (R 3.6.0)
 htmltools     0.4.0   2019-10-04 [1] CRAN (R 3.6.1)
 httpuv        1.5.2   2019-09-11 [1] CRAN (R 3.6.1)
 knitr         1.28    2020-02-06 [1] CRAN (R 3.6.2)
 later         1.0.0   2019-10-04 [1] CRAN (R 3.6.1)
 magrittr      1.5     2014-11-22 [1] CRAN (R 3.6.0)
 memoise       1.1.0   2017-04-21 [1] CRAN (R 3.6.0)
 pkgbuild      1.0.6   2019-10-09 [1] CRAN (R 3.6.1)
 pkgload       1.0.2   2018-10-29 [1] CRAN (R 3.6.0)
 prettyunits   1.1.1   2020-01-24 [1] CRAN (R 3.6.2)
 processx      3.4.2   2020-02-09 [1] CRAN (R 3.6.3)
 promises      1.1.0   2019-10-04 [1] CRAN (R 3.6.1)
 ps            1.3.2   2020-02-13 [1] CRAN (R 3.6.3)
 R6            2.4.1   2019-11-12 [1] CRAN (R 3.6.1)
 Rcpp          1.0.3   2019-11-08 [1] CRAN (R 3.6.1)
 remotes       2.1.1   2020-02-15 [1] CRAN (R 3.6.3)
 rlang         0.4.5   2020-03-01 [1] CRAN (R 3.6.3)
 rmarkdown     2.1     2020-01-20 [1] CRAN (R 3.6.2)
 rprojroot     1.3-2   2018-01-03 [1] CRAN (R 3.6.0)
 sessioninfo   1.1.1   2018-11-05 [1] CRAN (R 3.6.0)
 stringi       1.4.6   2020-02-17 [1] CRAN (R 3.6.3)
 stringr       1.4.0   2019-02-10 [1] CRAN (R 3.6.0)
 testthat      2.3.2   2020-03-02 [1] CRAN (R 3.6.3)
 usethis       1.5.1   2019-07-04 [1] CRAN (R 3.6.1)
 whisker       0.4     2019-08-28 [1] CRAN (R 3.6.1)
 withr         2.1.2   2018-03-15 [1] CRAN (R 3.6.0)
 workflowr   * 1.6.0   2019-12-19 [1] CRAN (R 3.6.2)
 xfun          0.12    2020-01-13 [1] CRAN (R 3.6.2)
 yaml          2.2.1   2020-02-01 [1] CRAN (R 3.6.2)

[1] /home/maimaizhu/R/x86_64-pc-linux-gnu-library/3.6
[2] /usr/local/lib/R/site-library
[3] /usr/lib/R/site-library
[4] /usr/lib/R/library</code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
