<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Xiang Zhu" />

<meta name="date" content="2019-08-18" />

<title>RSS-NET analysis of simulated GWAS summary statistics and B cell regulatory network</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/sandstone.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">rss-net</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="setup.html">Setup</a>
</li>
<li>
  <a href="function.html">Functions</a>
</li>
<li>
  <a href="faq.html">FAQ</a>
</li>
<li>
  <a href="https://stephenslab.github.io/rss/news.html">News</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/SUwonglab/rss-net">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">RSS-NET analysis of simulated GWAS summary statistics and B cell regulatory network</h1>
<h4 class="author">Xiang Zhu</h4>
<h4 class="date">2019-08-18</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-06-29
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>rss-net/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190823code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190823)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190823code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190823)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomSUwonglabrssnettreeddaf8781caef9bb4e6c66b105879c335b5afc360targetblankddaf878a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/SUwonglab/rss-net/tree/ddaf8781caef9bb4e6c66b105879c335b5afc360" target="_blank">ddaf878</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomSUwonglabrssnettreeddaf8781caef9bb4e6c66b105879c335b5afc360targetblankddaf878a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/SUwonglab/rss-net/tree/ddaf8781caef9bb4e6c66b105879c335b5afc360" target="_blank">ddaf878</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    examples/ibd2015_nkcell/ibd2015_Primary_Natural_Killer_cells_from_peripheral_blood_NK_out_66.mat
    Ignored:    examples/ibd2015_nkcell/randinit_results/
    Ignored:    examples/ibd2015_nkcell/results/
    Ignored:    examples/wtccc_bcell/results/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>rmd/wtccc_bcell.Rmd</code>) and HTML (<code>docs/wtccc_bcell.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/SUwonglab/rss-net/blob/ddaf8781caef9bb4e6c66b105879c335b5afc360/rmd/wtccc_bcell.Rmd" target="_blank">ddaf878</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-06-29
</td>
<td>
wflow_publish(“rmd/wtccc_bcell.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/SUwonglab/rss-net/93ce892f73c9b4f19d3bea1c6efb98fc5e5267c1/docs/wtccc_bcell.html" target="_blank">93ce892</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-06-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/SUwonglab/rss-net/260f36c1bd307d58edddbe9d3e4a58c718e1d50c/docs/wtccc_bcell.html" target="_blank">260f36c</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/SUwonglab/rss-net/blob/baf07eca495f8776475418a3feac069d62034810/rmd/wtccc_bcell.Rmd" target="_blank">baf07ec</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-15
</td>
<td>
wflow_publish(“rmd/wtccc_bcell.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/SUwonglab/rss-net/a4241718442f47362b6581697b185afd07f2a850/docs/wtccc_bcell.html" target="_blank">a424171</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/SUwonglab/rss-net/5c5e8e118c10fae0d8f22664a7dc383a24fe3406/docs/wtccc_bcell.html" target="_blank">5c5e8e1</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/SUwonglab/rss-net/31f90646bb8bc0a15754bd670a8d12c2bdddb0d0/docs/wtccc_bcell.html" target="_blank">31f9064</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/SUwonglab/rss-net/blob/9469f2d3f4d41e53ce5fdef7613825fe7a2b04bb/rmd/wtccc_bcell.Rmd" target="_blank">9469f2d</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-06
</td>
<td>
wflow_publish(“rmd/wtccc_bcell.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/SUwonglab/rss-net/f2c772c8c4c342450bfb8570c756cbc519450e39/docs/wtccc_bcell.html" target="_blank">f2c772c</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/SUwonglab/rss-net/blob/26bfe195ec9a50233df23fa2e2ab1542d32b4454/rmd/wtccc_bcell.Rmd" target="_blank">26bfe19</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-06
</td>
<td>
wflow_publish(“rmd/wtccc_bcell.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/SUwonglab/rss-net/991b3234f108057574d29bf37f84cf2a5306e74d/docs/wtccc_bcell.html" target="_blank">991b323</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/SUwonglab/rss-net/blob/e3daa54b1768e59db84e6861e18aff7c9c69c517/rmd/wtccc_bcell.Rmd" target="_blank">e3daa54</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-06
</td>
<td>
wflow_publish(“rmd/wtccc_bcell.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/SUwonglab/rss-net/3703a3df1c499e43d5cac80510e21d2a22a02cb3/docs/wtccc_bcell.html" target="_blank">3703a3d</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/SUwonglab/rss-net/blob/4de49b794924f2a8d5da69459f5a0975c928d663/rmd/wtccc_bcell.Rmd" target="_blank">4de49b7</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-06
</td>
<td>
wflow_publish(“rmd/wtccc_bcell.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/SUwonglab/rss-net/54677839aab0134f54fb1181ec52e30169a845d5/docs/wtccc_bcell.html" target="_blank">5467783</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/SUwonglab/rss-net/blob/9729b44e534c052483a62f63741e3829451ca473/rmd/wtccc_bcell.Rmd" target="_blank">9729b44</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-06
</td>
<td>
wflow_publish(“rmd/wtccc_bcell.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/SUwonglab/rss-net/c2ee73aba4f97f9331d8938b35c6c195b35e6599/docs/wtccc_bcell.html" target="_blank">c2ee73a</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/SUwonglab/rss-net/blob/b109ab5ad190bf8cb72e699d748909ff6fdaff80/rmd/wtccc_bcell.Rmd" target="_blank">b109ab5</a>
</td>
<td>
xiangzhu
</td>
<td>
2020-03-06
</td>
<td>
wflow_publish(“rmd/wtccc_bcell.Rmd”)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="overview" class="section level2">
<h2>Overview</h2>
<p>Here we describe an end-to-end RSS-NET analysis of two synthetic datasets that are simulated from real genotypes of 348965 genome-wide SNPs from 1458 individuals in the UK Blood Service Control Group (<a href="https://doi.org/10.1038/nature05911">Wellcome Trust Case Control Consortium, 2007</a>) and a gene regulatory network inferred for B cells <a href="https://doi.org/10.1038/nature11247">(ENCODE Project Consortium, 2012)</a>. This example gives a quick view of how RSS-NET works.</p>
<p>To reproduce results of this example, please use scripts in the directory <a href="https://github.com/SUwonglab/rss-net/tree/master/examples/wtccc_bcell"><code>wtccc_bcell/</code></a>, and follow the step-by-step guide below. Before running any script in <a href="https://github.com/SUwonglab/rss-net/tree/master/examples/wtccc_bcell"><code>wtccc_bcell/</code></a>, please <a href="setup.html">install</a> RSS-NET.</p>
<p>With the software installed and the input data downloaded, one should be able to run this example by simply typing the following line in shell:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode zsh"><code class="sourceCode zsh"><span id="cb1-1"><a href="#cb1-1"></a>$ sbatch run_simulation.sbatch</span></code></pre></div>
<p>If a different directory is used to store the software, input data and/or output data, please modify file paths in the given scripts accordingly.</p>
</div>
<div id="step-by-step-illustration" class="section level2">
<h2>Step-by-step illustration</h2>
<div id="download-data-files" class="section level3">
<h3>Download data files</h3>
<p>All data files required to run this example are freely available at Zenodo <a href="https://doi.org/10.5281/zenodo.3700214"><img src="https://zenodo.org/badge/DOI/10.5281/zenodo.3700214.svg" alt="DOI" /></a>. Please contact me (<code>xiangzhu[at]stanford.edu</code>) if you have any trouble accessing these files. After a complete download, you should see the following files.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode zsh"><code class="sourceCode zsh"><span id="cb2-1"><a href="#cb2-1"></a>$ tree <span class="kw">.</span></span>
<span id="cb2-2"><a href="#cb2-2"></a><span class="kw">.</span></span>
<span id="cb2-3"><a href="#cb2-3"></a>├── m0_data.mat</span>
<span id="cb2-4"><a href="#cb2-4"></a>├── m0_true_gene.mat</span>
<span id="cb2-5"><a href="#cb2-5"></a>├── m1_data.mat</span>
<span id="cb2-6"><a href="#cb2-6"></a>├── m1_true_gene.mat</span>
<span id="cb2-7"><a href="#cb2-7"></a>├── Primary_B_cells_from_peripheral_blood_gene2gene.mat</span>
<span id="cb2-8"><a href="#cb2-8"></a>├── wtccc_bcell_data.md5</span>
<span id="cb2-9"><a href="#cb2-9"></a>├── wtccc_bcell_results.zip</span>
<span id="cb2-10"><a href="#cb2-10"></a>├── WTCCC_Primary_B_cells_from_peripheral_blood_snp2gene_cis.mat</span>
<span id="cb2-11"><a href="#cb2-11"></a>├── WTCCC_Primary_B_cells_from_peripheral_blood_snp2net.mat</span>
<span id="cb2-12"><a href="#cb2-12"></a>├── WTCCC_snp2gene.mat</span>
<span id="cb2-13"><a href="#cb2-13"></a>└── wtccc_tidy_snp2gene_ncbi35togrch37.mat</span>
<span id="cb2-14"><a href="#cb2-14"></a></span>
<span id="cb2-15"><a href="#cb2-15"></a>0 directories, 11 files</span></code></pre></div>
<p>To help readers confirm if they use the same files as we do, we report 128-bit MD5 hashes of all files in <code>wtccc_bcell_data.md5</code>.</p>
<p>To help readers confirm if they can reproduce results of this example, we also provide the results (<code>wtccc_bcell_results.zip</code>) in the same Zenodo deposit.</p>
<p>For simplicity and generality, we introduce the following short-hand notations.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode matlab"><code class="sourceCode matlab"><span id="cb3-1"><a href="#cb3-1"></a>gwas = <span class="st">&#39;WTCCC&#39;</span>;</span>
<span id="cb3-2"><a href="#cb3-2"></a>net  = <span class="st">&#39;Primary_B_cells_from_peripheral_blood&#39;</span>; </span></code></pre></div>
<div id="m_data.mat-simulated-gwas-summary-statistics-and-reference-ld-estimates" class="section level4">
<h4>1. <code>m*_data.mat</code>: simulated GWAS summary statistics and reference LD estimates</h4>
<p>Here we simulate two synthetic datasets from WTCCC genotypes and B cell network: one (<code>m0_data.mat</code>) based on RSS-NET baseline model <span class="math inline">\(M_0\)</span>, and the other (<code>m1_data.mat</code>) based on RSS-NET enrichment model <span class="math inline">\(M_1\)</span>. We match <code>m*_data.mat</code> by the number of trait-associated SNPs and the proportion of phenotypic variation explained by all SNPs. By this design, <code>m*_data.mat</code> have the same amount of genetic signals, and they only differ in how genetic signals are distributed: for the enrichment dataset <code>m1_data.mat</code> genetic associations are enriched in the B cell network; for the baseline dataset <code>m0_data.mat</code> genetic associations are randomly distributed across the genome.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode zsh"><code class="sourceCode zsh"><span id="cb4-1"><a href="#cb4-1"></a>$ md5sum m*_data.mat</span>
<span id="cb4-2"><a href="#cb4-2"></a>5d652796b4cc358843920b1edf2819fc  m0_data.mat</span>
<span id="cb4-3"><a href="#cb4-3"></a>996eae1410f89607f1debefb31ec7811  m1_data.mat</span>
<span id="cb4-4"><a href="#cb4-4"></a></span>
<span id="cb4-5"><a href="#cb4-5"></a>$ <span class="kw">du</span> -sh m*_data.mat</span>
<span id="cb4-6"><a href="#cb4-6"></a>3.0G    m0_data.mat</span>
<span id="cb4-7"><a href="#cb4-7"></a>3.0G    m1_data.mat</span></code></pre></div>
<p>Let’s look at the contents of <code>m*_data.mat</code>.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode matlab"><code class="sourceCode matlab"><span id="cb5-1"><a href="#cb5-1"></a>&gt;&gt; sumstat=matfile(<span class="st">&#39;m0_data.mat&#39;</span>); </span>
<span id="cb5-2"><a href="#cb5-2"></a>&gt;&gt; sumstat</span>
<span id="cb5-3"><a href="#cb5-3"></a>  matlab.io.MatFile</span>
<span id="cb5-4"><a href="#cb5-4"></a>  Properties:</span>
<span id="cb5-5"><a href="#cb5-5"></a>      Properties.Source: <span class="st">&#39;m0_data.mat&#39;</span></span>
<span id="cb5-6"><a href="#cb5-6"></a>    Properties.Writable: false                                                  </span>
<span id="cb5-7"><a href="#cb5-7"></a>                  SiRiS: [<span class="fl">22</span>x1 cell]                                            </span>
<span id="cb5-8"><a href="#cb5-8"></a>                betahat: [<span class="fl">22</span>x1 cell]                                            </span>
<span id="cb5-9"><a href="#cb5-9"></a>                    chr: [<span class="fl">22</span>x1 cell]                                            </span>
<span id="cb5-10"><a href="#cb5-10"></a>                    pos: [<span class="fl">22</span>x1 cell]                                            </span>
<span id="cb5-11"><a href="#cb5-11"></a>                     se: [<span class="fl">22</span>x1 cell]</span></code></pre></div>
<p>GWAS summary statistics and LD estimates are stored as <a href="https://www.mathworks.com/help/matlab/cell-arrays.html">cell arrays</a>. RSS-NET only uses the following variables:</p>
<ul>
<li><code>betahat{j,1}</code>, single-SNP effect size estimates of all SNPs on chromosome <code>j</code>;</li>
<li><code>se{j,1}</code>, standard errors of <code>betahat{j, 1}</code>;</li>
<li><code>chr{j,1}</code> and <code>pos{j, 1}</code>, physical positions of these SNPs (GRCh37 build);</li>
<li><code>SiRiS{j,1}</code>, a <a href="https://www.mathworks.com/help/matlab/sparse-matrices.html">sparse matrix</a>, defined as <code>repmat((1./se),1,p) .* R .* repmat((1./se)',p,1)</code>, where <code>R</code> is the estimated LD matrix of these <code>p</code> SNPs.</li>
</ul>
</div>
<div id="gwas_snp2gene.mat-physical-distance-between-snps-and-genes" class="section level4">
<h4>2. <code>${gwas}_snp2gene.mat</code>: physical distance between SNPs and genes</h4>
<p>This file contains the physical distance between each GWAS SNP and each protein-coding gene, within 1 Mb. This file corresponds to <span class="math inline">\({\bf G}_j\)</span> in the RSS-NET model.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode zsh"><code class="sourceCode zsh"><span id="cb6-1"><a href="#cb6-1"></a>$ md5sum WTCCC_snp2gene.mat </span>
<span id="cb6-2"><a href="#cb6-2"></a>50da1887601df3a0914380e2ea9b1be7  WTCCC_snp2gene.mat</span>
<span id="cb6-3"><a href="#cb6-3"></a></span>
<span id="cb6-4"><a href="#cb6-4"></a>$ <span class="kw">du</span> -sh WTCCC_snp2gene.mat</span>
<span id="cb6-5"><a href="#cb6-5"></a>645M    WTCCC_snp2gene.mat</span></code></pre></div>
<p>In this example, there are 18334 genes and 348965 SNPs.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode matlab"><code class="sourceCode matlab"><span id="cb7-1"><a href="#cb7-1"></a>&gt;&gt; snp2gene=matfile(<span class="st">&#39;WTCCC_snp2gene.mat&#39;</span>);</span>
<span id="cb7-2"><a href="#cb7-2"></a>&gt;&gt; snp2gene</span>
<span id="cb7-3"><a href="#cb7-3"></a>  matlab.io.MatFile</span>
<span id="cb7-4"><a href="#cb7-4"></a>  Properties:</span>
<span id="cb7-5"><a href="#cb7-5"></a>      Properties.Source: <span class="st">&#39;WTCCC_snp2gene.mat&#39;</span></span>
<span id="cb7-6"><a href="#cb7-6"></a>    Properties.Writable: false                                                         </span>
<span id="cb7-7"><a href="#cb7-7"></a>                    chr: [<span class="fl">348965</span>x1   double]                                           </span>
<span id="cb7-8"><a href="#cb7-8"></a>                  colid: [<span class="fl">41897736</span>x1 int32]                                            </span>
<span id="cb7-9"><a href="#cb7-9"></a>             ncbi35_pos: [<span class="fl">348965</span>x1   int32]                                            </span>
<span id="cb7-10"><a href="#cb7-10"></a>                numgene: [<span class="fl">1</span>x1        int32]                                            </span>
<span id="cb7-11"><a href="#cb7-11"></a>                 numsnp: [<span class="fl">1</span>x1        int32]                                            </span>
<span id="cb7-12"><a href="#cb7-12"></a>                    pos: [<span class="fl">348965</span>x1   int32]                                            </span>
<span id="cb7-13"><a href="#cb7-13"></a>                  rowid: [<span class="fl">41897736</span>x1 int32]                                            </span>
<span id="cb7-14"><a href="#cb7-14"></a>                    val: [<span class="fl">41897736</span>x1 double]                                           </span>
<span id="cb7-15"><a href="#cb7-15"></a></span>
<span id="cb7-16"><a href="#cb7-16"></a>&gt;&gt; [snp2gene.numgene snp2gene.numsnp]</span>
<span id="cb7-17"><a href="#cb7-17"></a>    <span class="fl">18334</span>   <span class="fl">348965</span></span></code></pre></div>
<p>The SNP-to-gene distance information is captured by a three-column matrix <code>[colid rowid val]</code>. For example, the distance between gene <code>7</code> and SNP <code>161</code> is <code>2855196</code> bps.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode matlab"><code class="sourceCode matlab"><span id="cb8-1"><a href="#cb8-1"></a>&gt;&gt; colid=snp2gene.colid; rowid=snp2gene.rowid; val=snp2gene.val;</span>
<span id="cb8-2"><a href="#cb8-2"></a>&gt;&gt; [colid(<span class="fl">6688</span>) rowid(<span class="fl">6688</span>) val(<span class="fl">6688</span>)]</span>
<span id="cb8-3"><a href="#cb8-3"></a>         <span class="fl">7</span>       <span class="fl">161</span>   <span class="fl">2855196</span></span></code></pre></div>
</div>
<div id="net_gene2gene.mat-gene-regulatory-network" class="section level4">
<h4>3. <code>${net}_gene2gene.mat</code>: gene regulatory network</h4>
<p>This file contains information of gene-to-gene connections in a given regulatory network.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode zsh"><code class="sourceCode zsh"><span id="cb9-1"><a href="#cb9-1"></a>$ md5sum Primary_B_cells_from_peripheral_blood_gene2gene.mat</span>
<span id="cb9-2"><a href="#cb9-2"></a>0098dd6d85762b18d5d594da6c3cfe93  Primary_B_cells_from_peripheral_blood_gene2gene.mat</span>
<span id="cb9-3"><a href="#cb9-3"></a></span>
<span id="cb9-4"><a href="#cb9-4"></a>$ <span class="kw">du</span> -sh Primary_B_cells_from_peripheral_blood_gene2gene.mat</span>
<span id="cb9-5"><a href="#cb9-5"></a>1.8M    Primary_B_cells_from_peripheral_blood_gene2gene.mat</span></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="sourceCode matlab"><code class="sourceCode matlab"><span id="cb10-1"><a href="#cb10-1"></a>&gt;&gt; gene2gene=matfile(<span class="st">&#39;Primary_B_cells_from_peripheral_blood_gene2gene.mat&#39;</span>);</span>
<span id="cb10-2"><a href="#cb10-2"></a>&gt;&gt; gene2gene</span>
<span id="cb10-3"><a href="#cb10-3"></a>  matlab.io.MatFile</span>
<span id="cb10-4"><a href="#cb10-4"></a>  Properties:</span>
<span id="cb10-5"><a href="#cb10-5"></a>      Properties.Source: <span class="st">&#39;Primary_B_cells_from_peripheral_blood_gene2gene.mat&#39;</span></span>
<span id="cb10-6"><a href="#cb10-6"></a>    Properties.Writable: false                                                                                          </span>
<span id="cb10-7"><a href="#cb10-7"></a>                  colid: [<span class="fl">110062</span>x1 int32]                                                                               </span>
<span id="cb10-8"><a href="#cb10-8"></a>                numgene: [<span class="fl">1</span>x1      int32]                                                                               </span>
<span id="cb10-9"><a href="#cb10-9"></a>                  rowid: [<span class="fl">110062</span>x1 int32]                                                                               </span>
<span id="cb10-10"><a href="#cb10-10"></a>                    val: [<span class="fl">110062</span>x1 double]</span></code></pre></div>
<p>For implementation convenience, this file contains the trivial case where each gene is mapped to itself with <code>val=1</code>.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode matlab"><code class="sourceCode matlab"><span id="cb11-1"><a href="#cb11-1"></a>&gt;&gt; colid=gene2gene.colid; rowid=gene2gene.rowid; val=gene2gene.val;             </span>
<span id="cb11-2"><a href="#cb11-2"></a>&gt;&gt; [gene2gene.numgene sum(colid==rowid) unique(val(colid==rowid))]</span>
<span id="cb11-3"><a href="#cb11-3"></a>   <span class="fl">18334</span>   <span class="fl">18334</span>       <span class="fl">1</span></span></code></pre></div>
<p>For a given network, transcription factors (TFs) are stored in <code>rowid</code> and target genes (TGs) are stored in <code>colid</code>. In this example there are 3018 TGs and 436 TFs. Among these TFs and TGs, there are 91728 edges. The edge weights range from 0.61 to 1. These TF-to-TG connections and edge weights correspond to <span class="math inline">\(\{{\bf T}_g,v_{gt}\}\)</span> in the RSS-NET model.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode matlab"><code class="sourceCode matlab"><span id="cb12-1"><a href="#cb12-1"></a>&gt;&gt; [length(unique(colid(colid ~= rowid))) length(unique(rowid(colid ~= rowid)))]</span>
<span id="cb12-2"><a href="#cb12-2"></a>        <span class="fl">3018</span>         <span class="fl">436</span></span>
<span id="cb12-3"><a href="#cb12-3"></a></span>
<span id="cb12-4"><a href="#cb12-4"></a>&gt;&gt; [length(colid(colid ~= rowid)) length(rowid(colid ~= rowid))]</span>
<span id="cb12-5"><a href="#cb12-5"></a>       <span class="fl">91728</span>       <span class="fl">91728</span></span>
<span id="cb12-6"><a href="#cb12-6"></a></span>
<span id="cb12-7"><a href="#cb12-7"></a>&gt;&gt; val_tftg = val(colid ~= rowid);</span>
<span id="cb12-8"><a href="#cb12-8"></a>&gt;&gt; [min(val_tftg) quantile(val_tftg, <span class="fl">0.25</span>) median(val_tftg) quantile(val_tftg, <span class="fl">0.75</span>) max(val_tftg)]</span>
<span id="cb12-9"><a href="#cb12-9"></a>    <span class="fl">0.6101</span>    <span class="fl">0.6278</span>    <span class="fl">0.6515</span>    <span class="fl">0.6885</span>    <span class="fl">1.0000</span>      </span></code></pre></div>
</div>
<div id="gwas_net_snp2net.mat-snp-to-network-proximity-annotation" class="section level4">
<h4>4. <code>${gwas}_${net}_snp2net.mat</code>: SNP-to-network proximity annotation</h4>
<p>This file contains binary annotations whether a SNP is “near” the given network, that is, within 100 kb of any network element (TF, TG or associated regulatory elements).</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode zsh"><code class="sourceCode zsh"><span id="cb13-1"><a href="#cb13-1"></a>$ md5sum WTCCC_Primary_B_cells_from_peripheral_blood_snp2net.mat</span>
<span id="cb13-2"><a href="#cb13-2"></a>701264f811c0f5edd582ee3eec5cfd15  WTCCC_Primary_B_cells_from_peripheral_blood_snp2net.mat</span>
<span id="cb13-3"><a href="#cb13-3"></a></span>
<span id="cb13-4"><a href="#cb13-4"></a>$ <span class="kw">du</span> -sh WTCCC_Primary_B_cells_from_peripheral_blood_snp2net.mat</span>
<span id="cb13-5"><a href="#cb13-5"></a>9.4M    WTCCC_Primary_B_cells_from_peripheral_blood_snp2net.mat</span></code></pre></div>
<p>In this example, there are 348965 GWAS SNPs and 121308 of them are near the B cell network (i.e. <code>val=1</code>).</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode matlab"><code class="sourceCode matlab"><span id="cb14-1"><a href="#cb14-1"></a>&gt;&gt; snp2net=matfile(<span class="st">&#39;WTCCC_Primary_B_cells_from_peripheral_blood_snp2net.mat&#39;</span>);</span>
<span id="cb14-2"><a href="#cb14-2"></a>&gt;&gt; snp2net</span>
<span id="cb14-3"><a href="#cb14-3"></a>  matlab.io.MatFile</span>
<span id="cb14-4"><a href="#cb14-4"></a>  Properties:</span>
<span id="cb14-5"><a href="#cb14-5"></a>      Properties.Source: <span class="st">&#39;WTCCC_Primary_B_cells_from_peripheral_blood_snp2net.mat&#39;</span></span>
<span id="cb14-6"><a href="#cb14-6"></a>    Properties.Writable: false                                                                                              </span>
<span id="cb14-7"><a href="#cb14-7"></a>                    chr: [<span class="fl">348965</span>x1 double]                                                                                  </span>
<span id="cb14-8"><a href="#cb14-8"></a>             ncbi35_pos: [<span class="fl">348965</span>x1 int32]                                                                                   </span>
<span id="cb14-9"><a href="#cb14-9"></a>                    pos: [<span class="fl">348965</span>x1 int32]                                                                                   </span>
<span id="cb14-10"><a href="#cb14-10"></a>                  snpid: [<span class="fl">348965</span>x1 int32]                                                                                   </span>
<span id="cb14-11"><a href="#cb14-11"></a>                    val: [<span class="fl">348965</span>x1 double]                                                                                  </span>
<span id="cb14-12"><a href="#cb14-12"></a>                 window: [<span class="fl">1</span>x1      double]                                                                                  </span>
<span id="cb14-13"><a href="#cb14-13"></a></span>
<span id="cb14-14"><a href="#cb14-14"></a>&gt;&gt; [length(snp2net.val) sum(snp2net.val) snp2net.window]</span>
<span id="cb14-15"><a href="#cb14-15"></a>      <span class="fl">348965</span>      <span class="fl">121308</span>      <span class="fl">100000</span></span>
<span id="cb14-16"><a href="#cb14-16"></a></span>
<span id="cb14-17"><a href="#cb14-17"></a>&gt;&gt; unique(snp2net.val)&#39;</span>
<span id="cb14-18"><a href="#cb14-18"></a>     <span class="fl">0</span>     <span class="fl">1</span></span></code></pre></div>
</div>
<div id="gwas_net_snp2gene_cis.mat-snp-to-gene-cis-regulation" class="section level4">
<h4>5. <code>${gwas}_${net}_snp2gene_cis.mat</code>: SNP-to-gene cis regulation</h4>
<p>This file contains the SNP-to-gene cis regulation scores. (For simplicity we do not use context-matching cis eQTL studies in our simulations.) This file corresponds to <span class="math inline">\((c_{jg}-1)\)</span> in the RSS-NET model.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode zsh"><code class="sourceCode zsh"><span id="cb15-1"><a href="#cb15-1"></a>$ md5sum WTCCC_Primary_B_cells_from_peripheral_blood_snp2gene_cis.mat</span>
<span id="cb15-2"><a href="#cb15-2"></a>b77eebd576eee6d19ac3c8ed94681f00  WTCCC_Primary_B_cells_from_peripheral_blood_snp2gene_cis.mat</span>
<span id="cb15-3"><a href="#cb15-3"></a></span>
<span id="cb15-4"><a href="#cb15-4"></a>$ <span class="kw">du</span> -sh WTCCC_Primary_B_cells_from_peripheral_blood_snp2gene_cis.mat</span>
<span id="cb15-5"><a href="#cb15-5"></a>4.5M    WTCCC_Primary_B_cells_from_peripheral_blood_snp2gene_cis.mat</span></code></pre></div>
<div class="sourceCode" id="cb16"><pre class="sourceCode matlab"><code class="sourceCode matlab"><span id="cb16-1"><a href="#cb16-1"></a>&gt;&gt; snp2gene_cis = matfile(<span class="st">&#39;WTCCC_Primary_B_cells_from_peripheral_blood_snp2gene_cis.mat&#39;</span>);</span>
<span id="cb16-2"><a href="#cb16-2"></a>&gt;&gt; snp2gene_cis</span>
<span id="cb16-3"><a href="#cb16-3"></a>  matlab.io.MatFile</span>
<span id="cb16-4"><a href="#cb16-4"></a>  Properties:</span>
<span id="cb16-5"><a href="#cb16-5"></a>      Properties.Source: <span class="st">&#39;WTCCC_Primary_B_cells_from_peripheral_blood_snp2gene_cis.mat&#39;</span></span>
<span id="cb16-6"><a href="#cb16-6"></a>    Properties.Writable: false                                                                                                   </span>
<span id="cb16-7"><a href="#cb16-7"></a>                  colid: [<span class="fl">291181</span>x1 int32]                                                                                        </span>
<span id="cb16-8"><a href="#cb16-8"></a>                     l0: [<span class="fl">1</span>x1      double]                                                                                       </span>
<span id="cb16-9"><a href="#cb16-9"></a>                     l1: [<span class="fl">1</span>x1      double]                                                                                       </span>
<span id="cb16-10"><a href="#cb16-10"></a>                numgene: [<span class="fl">1</span>x1      int32]                                                                                        </span>
<span id="cb16-11"><a href="#cb16-11"></a>                 numsnp: [<span class="fl">1</span>x1      int32]                                                                                        </span>
<span id="cb16-12"><a href="#cb16-12"></a>                  rowid: [<span class="fl">291181</span>x1 int32]                                                                                        </span>
<span id="cb16-13"><a href="#cb16-13"></a>                    val: [<span class="fl">291181</span>x1 double]                                                                                       </span>
<span id="cb16-14"><a href="#cb16-14"></a>                 window: [<span class="fl">1</span>x1      double]</span></code></pre></div>
<p>In this example, there are 291181 SNP-gene pairs with cis regulation scores available, consisting of 111097 SNPs and 5345 genes. The cis regulation scores (<code>val</code>) range from 0.02 to 0.90.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode matlab"><code class="sourceCode matlab"><span id="cb17-1"><a href="#cb17-1"></a>&gt;&gt; [snp2gene_cis.numsnp length(unique(snp2gene_cis.rowid)) snp2gene_cis.numgene length(unique(snp2gene_cis.colid))]</span>
<span id="cb17-2"><a href="#cb17-2"></a>   <span class="fl">348965</span>   <span class="fl">111097</span>    <span class="fl">18334</span>     <span class="fl">5345</span></span>
<span id="cb17-3"><a href="#cb17-3"></a></span>
<span id="cb17-4"><a href="#cb17-4"></a>&gt;&gt; val=snp2gene_cis.val;</span>
<span id="cb17-5"><a href="#cb17-5"></a>&gt;&gt; [min(val) quantile(val, <span class="fl">0.25</span>) median(val) quantile(val, <span class="fl">0.75</span>) max(val)]</span>
<span id="cb17-6"><a href="#cb17-6"></a>    <span class="fl">0.0202</span>    <span class="fl">0.1615</span>    <span class="fl">0.2661</span>    <span class="fl">0.3864</span>    <span class="fl">0.8963</span></span></code></pre></div>
</div>
</div>
<div id="run-rss-net-analysis" class="section level3">
<h3>Run RSS-NET analysis</h3>
<p>To facilitate running RSS-NET on real data, we provide a generic script <a href="https://github.com/SUwonglab/rss-net/tree/master/examples/wtccc_bcell/simulation_template.m"><code>simulation_template.m</code></a>. For the present example, the RSS-NET analysis of two synthetic datasets is implemented by <a href="https://github.com/SUwonglab/rss-net/tree/master/examples/wtccc_bcell/run_simulation.m"><code>run_simulation.m</code></a> and <a href="https://github.com/SUwonglab/rss-net/tree/master/examples/wtccc_bcell/run_simulation.sbatch"><code>run_simulation.sbatch</code></a>.</p>
<p>Please see scripts in <a href="https://github.com/SUwonglab/rss-net/tree/master/examples/wtccc_bcell"><code>wtccc_bcell/</code></a> for full details. Below we only highlight the important parts.</p>
<div id="specify-input-data-files" class="section level4">
<h4>1. Specify input data files</h4>
<p>First we need to tell RSS-NET where to find all required input files, by creating the following <a href="https://www.mathworks.com/help/matlab/ref/struct.html">structure array</a> <code>data</code>. Please ensure that you save all downloaded files in <code>dat_path</code>.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode matlab"><code class="sourceCode matlab"><span id="cb18-1"><a href="#cb18-1"></a>data.sumstats_file     = strcat(dat_path,trial_name,<span class="st">&#39;_data.mat&#39;</span>);</span>
<span id="cb18-2"><a href="#cb18-2"></a>data.snp2net_file      = strcat(dat_path,gwas_name,<span class="st">&#39;_&#39;</span>,net_name,<span class="st">&#39;_snp2net.mat&#39;</span>);</span>
<span id="cb18-3"><a href="#cb18-3"></a>data.snp2gene_file     = strcat(dat_path,gwas_name,<span class="st">&#39;_snp2gene.mat&#39;</span>);</span>
<span id="cb18-4"><a href="#cb18-4"></a>data.gene2gene_file    = strcat(dat_path,net_name,<span class="st">&#39;_gene2gene.mat&#39;</span>);</span>
<span id="cb18-5"><a href="#cb18-5"></a>data.snp2gene_cis_file = strcat(dat_path,gwas_name,<span class="st">&#39;_&#39;</span>,net_name,<span class="st">&#39;_snp2gene_cis.mat&#39;</span>);</span></code></pre></div>
</div>
<div id="specify-a-hyper-parameter-grid" class="section level4">
<h4>2. Specify a hyper-parameter grid</h4>
<p>Next we need feed a hyper-parameter grid to RSS-NET, by creating the following <a href="https://www.mathworks.com/help/matlab/ref/struct.html">structure array</a> <code>hyper</code>. The total computational cost of RSS-NET is proportional to the grid size. Hence, if you want to finish running this example faster, you can always reduce the grid size.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode matlab"><code class="sourceCode matlab"><span id="cb19-1"><a href="#cb19-1"></a>hyper.theta0 = true_theta0;</span>
<span id="cb19-2"><a href="#cb19-2"></a>hyper.sigma0 = true_sigma0;</span>
<span id="cb19-3"><a href="#cb19-3"></a>hyper.theta  = [<span class="fl">0</span> ((true_theta-<span class="fl">0.5</span>):<span class="fl">0.25</span>:(true_theta+<span class="fl">0.5</span>))];</span>
<span id="cb19-4"><a href="#cb19-4"></a>hyper.sigma  = [<span class="fl">0</span> ((true_sigma-<span class="fl">0.5</span>):<span class="fl">0.25</span>:(true_sigma+<span class="fl">0.5</span>))];</span></code></pre></div>
</div>
<div id="initialize-variational-parameters" class="section level4">
<h4>3. Initialize variational parameters</h4>
<p>We can provide initial values for variational parameters in RSS-NET as follows. If we do not provide initial values, RSS-NET will use a random initialization.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode matlab"><code class="sourceCode matlab"><span id="cb20-1"><a href="#cb20-1"></a>options.alpha = alpha0;</span>
<span id="cb20-2"><a href="#cb20-2"></a>options.mu    = mu0;</span></code></pre></div>
</div>
<div id="fit-the-rss-net-model" class="section level4">
<h4>4. Fit the RSS-NET model</h4>
<p>We use <a href="https://github.com/SUwonglab/rss-net/blob/master/src/rss_net.m"><code>rss_net.m</code></a> to fit the RSS-NET model on a given dataset. The model fitting results consist of variational lower bounds <code>logw</code> and variational parameter estimates <code>[alpha mu s]</code> for the given hyper-parameter grid.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode matlab"><code class="sourceCode matlab"><span id="cb21-1"><a href="#cb21-1"></a>[logw,alpha,mu,s,param] = rss_net(data,hyper,options);</span></code></pre></div>
</div>
<div id="summarize-network-level-enrichments" class="section level4">
<h4>5. Summarize network-level enrichments</h4>
<p>To assess whether a regulatory network is enriched for genetic associations with a trait, we evaluate a Bayes factor (BF) comparing the baseline model (<span class="math inline">\(M_0:\theta=0~\text{and}~\sigma^2=0\)</span>) in RSS-NET with the enrichment model (<span class="math inline">\(M_1:\theta&gt;0~\text{or}~\sigma^2&gt;0\)</span>). The BF computation is implemented in <a href="https://github.com/SUwonglab/rss-net/blob/master/src/calc_bf.m"><code>calc_bf.m</code></a>.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode matlab"><code class="sourceCode matlab"><span id="cb22-1"><a href="#cb22-1"></a>log10_bf = calc_bf(logw,param);</span></code></pre></div>
</div>
<div id="summarize-gene-level-associations" class="section level4">
<h4>6. Summarize gene-level associations</h4>
<p>To summarize association between a locus and a trait, we compute <span class="math inline">\(P_1^{\sf net}\)</span>, the posterior probability that at least one SNP <span class="math inline">\(j\)</span> in the locus is associated with the trait (<span class="math inline">\(\beta_j\neq 0\)</span>): <span class="math display">\[
P_1^{\sf net}=1-\Pr(\beta_j=0,~\forall j\in\text{locus}~|~\text{data},M_1).
\]</span> As in <a href="https://doi.org/10.1101/2020.03.13.990010">Zhu et al (2020)</a>, a locus is defined as the transcribed region of a gene plus 100 kb upstream and downstream. The locus definition is provided in <code>wtccc_tidy_snp2gene_ncbi35togrch37.mat</code>. The <span class="math inline">\(P_1^{\sf net}\)</span> computation is implemented in <a href="https://github.com/stephenslab/rss/blob/master/src_vb/compute_pip.m"><code>compute_pip.m</code></a>.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode matlab"><code class="sourceCode matlab"><span id="cb23-1"><a href="#cb23-1"></a>seg_path = strcat(dat_path,<span class="st">&#39;wtccc_tidy_snp2gene_ncbi35togrch37.mat&#39;</span>);</span>
<span id="cb23-2"><a href="#cb23-2"></a>p1_net   = compute_pip(seg_path,logw,alpha);</span></code></pre></div>
</div>
<div id="submit-jobs" class="section level4">
<h4>7. Submit jobs</h4>
<p>After understanding the basic workflow of RSS-NET, we submit two jobs of running RSS-NET on the synthetic datasets <code>m*_data.mat</code>. To this end, we write a simple sbatch script <a href="https://github.com/SUwonglab/rss-net/tree/master/examples/wtccc_bcell/run_simulation.sbatch"><code>run_simulation.sbatch</code></a>, and submit it to a cluster with <a href="https://slurm.schedmd.com/"><code>Slurm</code></a> available.</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode zsh"><code class="sourceCode zsh"><span id="cb24-1"><a href="#cb24-1"></a>$ squeue -u xiangzhu</span>
<span id="cb24-2"><a href="#cb24-2"></a>             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST<span class="kw">(</span>REASON<span class="kw">)</span></span>
<span id="cb24-3"><a href="#cb24-3"></a>        66258089_0   broadwl run_simu xiangzhu  R    2:48:17      1 midway2-0009</span>
<span id="cb24-4"><a href="#cb24-4"></a>        66258089_1   broadwl run_simu xiangzhu  R    2:48:17      1 midway2-0011</span></code></pre></div>
<p>For each job, we request 1 node with 8 CPUs and 32 Gb total memory. The memory utilized by these two jobs is around 16 Gb each. The running time of these two jobs is around 4 hours each.</p>
</div>
<div id="understand-analysis-results" class="section level4">
<h4>8. Understand analysis results</h4>
<p>Running <a href="https://github.com/SUwonglab/rss-net/tree/master/examples/wtccc_bcell/run_simulation.sbatch"><code>run_simulation.sbatch</code></a> yields two MAT-files <code>m*_results.mat</code>.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode matlab"><code class="sourceCode matlab"><span id="cb25-1"><a href="#cb25-1"></a>&gt;&gt; m0_res=matfile(<span class="st">&#39;m0_results.mat&#39;</span>);</span>
<span id="cb25-2"><a href="#cb25-2"></a>&gt;&gt; m1_res=matfile(<span class="st">&#39;m1_results.mat&#39;</span>);</span></code></pre></div>
<p>As shown below, RSS-NET yields a small BF (<span class="math inline">\(0.13\)</span>) for the baseline dataset <code>m0_data.mat</code> and a large BF (<span class="math inline">\(5.85\times 10^{55}\)</span>) for the enrichment dataset <code>m1_data.mat</code>.</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode matlab"><code class="sourceCode matlab"><span id="cb26-1"><a href="#cb26-1"></a>&gt;&gt; [m0_res.log10_bf m1_res.log10_bf]</span>
<span id="cb26-2"><a href="#cb26-2"></a>   -<span class="fl">0.8917</span>   <span class="fl">55.7671</span></span></code></pre></div>
<p>We also compare gene-level associations inferred from RSS-NET (<code>m*_res.pp</code>) with ground truth (<code>m*_true_gene.mat</code>). RSS-NET shows a higher statistical power in identifying gene-level associations on the enrichment dataset <code>m1_data.mat</code> (<span class="math inline">\(\text{AUC}=0.78\)</span>) than the baseline dataset <code>m0_data.mat</code> (<span class="math inline">\(\text{AUC}=0.60\)</span>). This is because RSS-NET automatically exploits the underlying network enrichment in <code>m1_data.mat</code> when assessing genetic associations.</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode matlab"><code class="sourceCode matlab"><span id="cb27-1"><a href="#cb27-1"></a>&gt;&gt; m1_gene=matfile(<span class="st">&#39;m1_true_gene.mat&#39;</span>);              </span>
<span id="cb27-2"><a href="#cb27-2"></a>&gt;&gt; m1_mdl = fitglm(m1_res.pp,m1_gene.true_gene,<span class="st">&#39;Distribution&#39;</span>,<span class="st">&#39;binomial&#39;</span>,<span class="st">&#39;Link&#39;</span>,<span class="st">&#39;logit&#39;</span>);</span>
<span id="cb27-3"><a href="#cb27-3"></a>&gt;&gt; [X,Y,T,m1_auc] = perfcurve(m1_gene.true_gene,m1_mdl.Fitted.Probability,<span class="fl">1</span>);</span>
<span id="cb27-4"><a href="#cb27-4"></a>&gt;&gt; m1_auc</span>
<span id="cb27-5"><a href="#cb27-5"></a>    <span class="fl">0.7767</span></span>
<span id="cb27-6"><a href="#cb27-6"></a></span>
<span id="cb27-7"><a href="#cb27-7"></a>&gt;&gt; m0_gene=matfile(<span class="st">&#39;m0_true_gene.mat&#39;</span>);                                                  </span>
<span id="cb27-8"><a href="#cb27-8"></a>&gt;&gt; m0_mdl = fitglm(m0_res.pp,m0_gene.true_gene,<span class="st">&#39;Distribution&#39;</span>,<span class="st">&#39;binomial&#39;</span>,<span class="st">&#39;Link&#39;</span>,<span class="st">&#39;logit&#39;</span>);</span>
<span id="cb27-9"><a href="#cb27-9"></a>&gt;&gt; [X,Y,T,m0_auc] = perfcurve(m0_gene.true_gene,m0_mdl.Fitted.Probability,<span class="fl">1</span>);</span>
<span id="cb27-10"><a href="#cb27-10"></a>&gt;&gt; m0_auc</span>
<span id="cb27-11"><a href="#cb27-11"></a>    <span class="fl">0.5954</span></span></code></pre></div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1"></a>devtools<span class="op">::</span><span class="kw">session_info</span>()</span></code></pre></div>
<pre><code>─ Session info ───────────────────────────────────────────────────────────────
 setting  value                       
 version  R version 4.0.2 (2020-06-22)
 os       Ubuntu 20.04 LTS            
 system   x86_64, linux-gnu           
 ui       X11                         
 language (EN)                        
 collate  en_US.UTF-8                 
 ctype    en_US.UTF-8                 
 tz       America/Los_Angeles         
 date     2020-06-29                  

─ Packages ───────────────────────────────────────────────────────────────────
 package     * version date       lib source        
 assertthat    0.2.1   2019-03-21 [3] CRAN (R 4.0.0)
 backports     1.1.8   2020-06-17 [3] CRAN (R 4.0.1)
 callr         3.4.3   2020-03-28 [1] CRAN (R 4.0.2)
 cli           2.0.2   2020-02-28 [3] CRAN (R 4.0.0)
 crayon        1.3.4   2017-09-16 [3] CRAN (R 4.0.0)
 desc          1.2.0   2018-05-01 [3] CRAN (R 4.0.0)
 devtools      2.3.0   2020-04-10 [3] CRAN (R 4.0.0)
 digest        0.6.25  2020-02-23 [3] CRAN (R 4.0.0)
 ellipsis      0.3.1   2020-05-15 [3] CRAN (R 4.0.0)
 evaluate      0.14    2019-05-28 [3] CRAN (R 4.0.0)
 fansi         0.4.1   2020-01-08 [3] CRAN (R 4.0.0)
 fs            1.4.1   2020-04-04 [3] CRAN (R 4.0.0)
 git2r         0.27.1  2020-05-03 [3] CRAN (R 4.0.0)
 glue          1.4.1   2020-05-13 [3] CRAN (R 4.0.0)
 htmltools     0.5.0   2020-06-16 [3] CRAN (R 4.0.1)
 httpuv        1.5.4   2020-06-06 [3] CRAN (R 4.0.0)
 knitr         1.29    2020-06-23 [3] CRAN (R 4.0.2)
 later         1.1.0.1 2020-06-05 [3] CRAN (R 4.0.0)
 magrittr      1.5     2014-11-22 [3] CRAN (R 4.0.0)
 memoise       1.1.0   2017-04-21 [3] CRAN (R 4.0.0)
 pkgbuild      1.0.8   2020-05-07 [3] CRAN (R 4.0.0)
 pkgload       1.1.0   2020-05-29 [3] CRAN (R 4.0.0)
 prettyunits   1.1.1   2020-01-24 [3] CRAN (R 4.0.0)
 processx      3.4.2   2020-02-09 [3] CRAN (R 4.0.0)
 promises      1.1.1   2020-06-09 [3] CRAN (R 4.0.1)
 ps            1.3.3   2020-05-08 [3] CRAN (R 4.0.0)
 R6            2.4.1   2019-11-12 [3] CRAN (R 4.0.0)
 Rcpp          1.0.4.6 2020-04-09 [3] CRAN (R 4.0.0)
 remotes       2.1.1   2020-02-15 [3] CRAN (R 4.0.0)
 rlang         0.4.6   2020-05-02 [3] CRAN (R 4.0.0)
 rmarkdown     2.3     2020-06-18 [1] CRAN (R 4.0.2)
 rprojroot     1.3-2   2018-01-03 [3] CRAN (R 4.0.0)
 sessioninfo   1.1.1   2018-11-05 [3] CRAN (R 4.0.0)
 stringi       1.4.6   2020-02-17 [3] CRAN (R 4.0.0)
 stringr       1.4.0   2019-02-10 [3] CRAN (R 4.0.0)
 testthat      2.3.2   2020-03-02 [3] CRAN (R 4.0.0)
 usethis       1.6.1   2020-04-29 [3] CRAN (R 4.0.0)
 whisker       0.4     2019-08-28 [3] CRAN (R 4.0.0)
 withr         2.2.0   2020-04-20 [3] CRAN (R 4.0.0)
 workflowr   * 1.6.2   2020-04-30 [1] CRAN (R 4.0.0)
 xfun          0.15    2020-06-21 [3] CRAN (R 4.0.1)
 yaml          2.2.1   2020-02-01 [3] CRAN (R 4.0.0)

[1] /home/maimaizhu/R/x86_64-pc-linux-gnu-library/4.0
[2] /usr/local/lib/R/site-library
[3] /usr/lib/R/site-library
[4] /usr/lib/R/library</code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
